Index: redis-2.2.7/src/sha1.c
===================================================================
--- redis-2.2.7.orig/src/sha1.c	2011-05-20 19:48:19.447230461 +0400
+++ redis-2.2.7/src/sha1.c	2011-05-20 19:48:46.379349364 +0400
@@ -41,6 +41,7 @@
 #define	PDP_ENDIAN	3412	/* LSB first in word, MSW first in long (pdp)*/
 
 #if defined(vax) || defined(ns32000) || defined(sun386) || defined(__i386__) || \
+    defined(__amd64__) || \
     defined(MIPSEL) || defined(_MIPSEL) || defined(BIT_ZERO_ON_RIGHT) || \
     defined(__alpha__) || defined(__alpha)
 #define BYTE_ORDER	LITTLE_ENDIAN
