This is a distro patch that should go upstream.

Index: python-dns-2.3.4/tests/test5.py
===================================================================
--- python-dns-2.3.4.orig/tests/test5.py	2010-06-22 19:42:02.511828022 -0400
+++ python-dns-2.3.4/tests/test5.py	2010-06-22 19:44:04.687816864 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python/
+#!/usr/bin/python
 
 import sys ; sys.path.insert(0, '..')
 
Index: python-dns-2.3.4/tests/testPackers.py
===================================================================
--- python-dns-2.3.4.orig/tests/testPackers.py	2010-06-22 19:42:41.299817420 -0400
+++ python-dns-2.3.4/tests/testPackers.py	2010-06-22 19:44:04.691817302 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 #
 # Tests of the packet assembler/disassembler routines.
Index: python-dns-2.3.4/tests/testsrv.py
===================================================================
--- python-dns-2.3.4.orig/tests/testsrv.py	2010-06-22 19:43:13.818817253 -0400
+++ python-dns-2.3.4/tests/testsrv.py	2010-06-22 19:44:04.691817302 -0400
@@ -1,4 +1,4 @@
-#!usr/bin/python
+#!/usr/bin/python
 
 import sys ; sys.path.insert(0, '..')
 
Index: python-dns-2.3.4/tools/named-perf.py
===================================================================
--- python-dns-2.3.4.orig/tools/named-perf.py	2010-06-22 19:43:34.207819339 -0400
+++ python-dns-2.3.4/tools/named-perf.py	2010-06-22 19:44:04.691817302 -0400
@@ -1,4 +1,4 @@
-#!/opt/python/bin/python
+#!/usr/bin/python
 
 servers = [ "192.92.129.1",
 	"192.189.54.17", # yarrina
