Index: ntp_4.2.5p200/include/ntp_config.h
===================================================================
--- ntp_4.2.5p200.orig/include/ntp_config.h
+++ ntp_4.2.5p200/include/ntp_config.h
@@ -10,7 +10,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE	"%windir%\\system32\\drivers\\etc\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.conf"
Index: ntp_4.2.5p200/libntp/audio.c
===================================================================
--- ntp_4.2.5p200.orig/libntp/audio.c
+++ ntp_4.2.5p200/libntp/audio.c
@@ -53,7 +53,7 @@
 static struct audio_device device; /* audio device ident */
 #endif /* HAVE_SYS_AUDIOIO_H */
 #ifdef PCM_STYLE_SOUND
-# define INIT_FILE "/etc/ntp.audio"
+# define INIT_FILE "/etc/inet/ntp.audio"
 int agc =	SOUND_MIXER_WRITE_RECLEV; /* or IGAIN or LINE */
 int monitor =	SOUND_MIXER_WRITE_VOLUME; /* or OGAIN */
 int devmask = 0;
Index: ntp_4.2.5p200/ntpd/refclock_oncore.c
===================================================================
--- ntp_4.2.5p200.orig/ntpd/refclock_oncore.c
+++ ntp_4.2.5p200/ntpd/refclock_oncore.c
@@ -1188,7 +1188,7 @@ oncore_read_config(
 
 	FILE	*fd;
 	char	*cp, *cc, *ca, line[100], units[2], device[20], Msg[160], **cpp;
-	char	*dirs[] = { "/etc/ntp", "/etc", 0 };
+	char	*dirs[] = { "/etc/inet", "/etc", 0 };
 	int	i, sign, lat_flg, long_flg, ht_flg, mode, mask;
 	double	f1, f2, f3;
 
Index: ntp_4.2.5p200/ntpdate/ntpdate.c
===================================================================
--- ntp_4.2.5p200.orig/ntpdate/ntpdate.c
+++ ntp_4.2.5p200/ntpdate/ntpdate.c
@@ -192,7 +192,7 @@ int complete_servers = 0;
 
 #ifndef KEYFILE
 # ifndef SYS_WINNT
-#define KEYFILE 	"/etc/ntp.keys"
+#define KEYFILE 	"/etc/inet/ntp.keys"
 # else
 #define KEYFILE 	"%windir%\\ntp.keys"
 # endif /* SYS_WINNT */
Index: ntp_4.2.5p200/ntpdate/ntptime_config.c
===================================================================
--- ntp_4.2.5p200.orig/ntpdate/ntptime_config.c
+++ ntp_4.2.5p200/ntpdate/ntptime_config.c
@@ -37,7 +37,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE 	"%windir%\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.ini"
Index: ntp_4.2.5p200/ntpdate/ntptimeset.c
===================================================================
--- ntp_4.2.5p200.orig/ntpdate/ntptimeset.c
+++ ntp_4.2.5p200/ntpdate/ntptimeset.c
@@ -270,7 +270,7 @@ u_long current_time = 0;
 
 #ifndef KEYFILE
 # ifndef SYS_WINNT
-#define KEYFILE 	"/etc/ntp.keys"
+#define KEYFILE 	"/etc/inet/ntp.keys"
 # else
 #define KEYFILE 	"%windir%\\ntp.keys"
 # endif /* SYS_WINNT */
Index: ntp_4.2.5p200/sntp/header.h
===================================================================
--- ntp_4.2.5p200.orig/sntp/header.h
+++ ntp_4.2.5p200/sntp/header.h
@@ -19,10 +19,10 @@ No changes should be needed for any syst
 #define MAX_SOCKETS        10          /* Maximum number of addresses */
 
 #ifndef LOCKNAME
-#    define LOCKNAME "/etc/sntp.pid"  /* Stores the pid */
+#    define LOCKNAME "/var/run/sntp.pid"  /* Stores the pid */
 #endif
 #ifndef SAVENAME
-#    define SAVENAME "/etc/sntp.state" /* Stores the recovery state */
+#    define SAVENAME "/var/ntp/sntp.state" /* Stores the recovery state */
 #endif
 
 
