Index: tcl8.5.9/unix/configure
===================================================================
--- tcl8.5.9.orig/unix/configure
+++ tcl8.5.9/unix/configure
@@ -9259,7 +9259,8 @@ fi
 	    # symbols when dynamically loaded into tclsh.
 
 	    SHLIB_LD_LIBS='${LIBS}'
-	    SHLIB_SUFFIX=".so"
+	    SHLIB_SUFFIX=".so.0"
+	    TCL_SHLIB_LD_EXTRAS="-Wl,-soname,\${TCL_LIB_FILE}"
 	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-ldl"
 	    if test "$GCC" = yes; then
Index: tcl/tcl8.5.9/unix/tcl.m4
===================================================================
--- tcl8.5.9.orig/unix/tcl.m4
+++ tcl8.5.9/unix/tcl.m4
@@ -1984,7 +1984,8 @@ dnl AC_CHECK_TOOL(AR, ar)
 	    # symbols when dynamically loaded into tclsh.
 
 	    SHLIB_LD_LIBS='${LIBS}'
-	    SHLIB_SUFFIX=".so"
+	    SHLIB_SUFFIX=".so.0"
+	    TCL_SHLIB_LD_EXTRAS="-Wl,-soname,\${TCL_LIB_FILE}"
 	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-ldl"
 	    AS_IF([test "$GCC" = yes], [
