Index: pyzmq-2.1.7/setup.py
===================================================================
--- pyzmq-2.1.7.orig/setup.py	2011-08-15 18:59:35.931885625 +0400
+++ pyzmq-2.1.7/setup.py	2011-08-15 20:01:15.907766502 +0400
@@ -122,6 +122,10 @@
     if sys.platform.startswith('freebsd'):
         COMPILER_SETTINGS['libraries'].append('pthread')
     
+    # add libgcc_s on NCP4
+    if sys.platform.startswith('sunos5'):
+        COMPILER_SETTINGS['libraries'].append('gcc_s')
+
     if ZMQ is not None:
         COMPILER_SETTINGS['include_dirs'] += [pjoin(ZMQ, 'include')]
         COMPILER_SETTINGS['library_dirs'] += [pjoin(ZMQ, 'lib')]
