--- /dev/null	2006-10-18 16:35:35.000000000 +0100
+++ ft/conf.d/52-languageselector.conf	2006-11-28 17:22:00.000000000 +0000
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!--
+  /etc/fonts/conf.avail/52-languageselector.conf
+  Load local ubuntu-specific language custom file
+-->
+<fontconfig>
+       <include ignore_missing="yes">language-selector.conf</include>
+</fontconfig>
