Index: trousers-0.3.4/src/tspi/rpc/hosttable.c
===================================================================
--- trousers-0.3.4.orig/src/tspi/rpc/hosttable.c	2011-04-14 20:27:06.312262605 +0400
+++ trousers-0.3.4/src/tspi/rpc/hosttable.c	2011-04-07 14:42:58.760877464 +0400
@@ -21,7 +21,7 @@
 
 struct host_table *ht = NULL;
 
-TSS_RESULT
+static TSS_RESULT 
 host_table_init()
 {
 	ht = calloc(1, sizeof(struct host_table));
@@ -35,8 +35,7 @@
 	return TSS_SUCCESS;
 }
 #ifdef SOLARIS
-#pragma init(_init)
-void _init(void)
+static void my_init(void)
 #else
 void __attribute__ ((constructor)) my_init(void)
 #endif
@@ -45,8 +44,7 @@
 	__tspi_obj_list_init();
 }
 
-#if 0
-void
+static void
 host_table_final()
 {
 	struct host_table_entry *hte, *next = NULL;
@@ -56,6 +54,10 @@
 	for (hte = ht->entries; hte; hte = next) {
 		if (hte)
 			next = hte->next;
+		if (hte->hostname)
+			free(hte->hostname);
+		if (hte->comm.buf)
+			free(hte->comm.buf);
 		free(hte);
 	}
 
@@ -65,10 +67,20 @@
 	ht = NULL;
 }
 
+#ifdef SOLARIS
+static void my_fini(void)
+#else
 void __attribute__ ((destructor)) my_fini(void)
+#endif
 {
 	host_table_final();
 }
+
+#ifdef SOLARIS
+
+#pragma init(my_init)
+#pragma fini(my_fini)
+
 #endif
 
 TSS_RESULT
@@ -100,6 +112,7 @@
 		if (tmp->tspContext == tspContext) {
 			LogError("Tspi_Context_Connect attempted on an already connected context!");
 			MUTEX_UNLOCK(ht->lock);
+			free(entry->hostname);
 			free(entry->comm.buf);
 			free(entry);
 			return TSPERR(TSS_E_CONNECTION_FAILED);
@@ -133,6 +146,8 @@
 				prev->next = hte->next;
 			else
 				ht->entries = hte->next;
+			if (hte->hostname)
+				free(hte->hostname);
 			free(hte->comm.buf);
 			free(hte);
 			break;
