Index: trousers-0.3.4/src/include/tcsd.h
===================================================================
--- trousers-0.3.4.orig/src/include/tcsd.h	2011-04-07 01:39:19.190266876 +0400
+++ trousers-0.3.4/src/include/tcsd.h	2011-04-07 01:42:19.759664773 +0400
@@ -48,14 +48,23 @@
 							of this TCS System */
 };
 
+#ifdef SOLARIS
+#define TCSD_CONFIG_FILE	"/etc/security/tcsd.conf"
+#else
 #define TCSD_CONFIG_FILE	ETC_PREFIX "/tcsd.conf"
+#endif
 
 #define TSS_USER_NAME		"tss"
 #define TSS_GROUP_NAME		"tss"
 
 #define TCSD_DEFAULT_MAX_THREADS	10
+#ifdef SOLARIS
+#define TCSD_DEFAULT_SYSTEM_PS_FILE	"/var/tpm/system/system.data"
+#define TCSD_DEFAULT_SYSTEM_PS_DIR	"/var/tpm/system"
+#else
 #define TCSD_DEFAULT_SYSTEM_PS_FILE	VAR_PREFIX "/lib/tpm/system.data"
 #define TCSD_DEFAULT_SYSTEM_PS_DIR	VAR_PREFIX "/lib/tpm"
+#endif /* SOLARIS */
 #define TCSD_DEFAULT_FIRMWARE_LOG_FILE	"/sys/kernel/security/tpm0/binary_bios_measurements"
 #define TCSD_DEFAULT_KERNEL_LOG_FILE	"/sys/kernel/security/ima/binary_runtime_measurements"
 #define TCSD_DEFAULT_FIRMWARE_PCRS	0x00000000
