Index: bzr-2.4.0/bzrlib/tests/test_osutils.py
===================================================================
--- bzr-2.4.0.orig/bzrlib/tests/test_osutils.py	2011-08-23 00:23:16.135918018 +0400
+++ bzr-2.4.0/bzrlib/tests/test_osutils.py	2011-08-24 14:04:52.408083263 +0400
@@ -300,7 +300,8 @@
             if e.errno not in (errno.ENOENT,):
                 raise
         else:
-            self.assertEquals('chardev', osutils.file_kind('/dev/null'))
+            dev_null_kind = osutils.file_kind(os.path.realpath('/dev/null'))
+            self.assertEquals('chardev', dev_null_kind)
 
         mkfifo = getattr(os, 'mkfifo', None)
         if mkfifo:
