diff -r 559e9710315c -r 789cee7bf738 usr/src/lib/pylibbe/common/libbe_py.c
--- a/usr/src/lib/pylibbe/common/libbe_py.c	Wed Jan 11 02:37:31 2012 -0800
+++ b/usr/src/lib/pylibbe/common/libbe_py.c	Tue Jan 31 04:58:36 2012 -0800
@@ -31,6 +31,10 @@
 #include <libbe.h>
 #include <libbe_priv.h>
 
+#define B_TRUE _B_TRUE
+#define B_FALSE _B_FALSE
+
+
 enum {
 	BE_PY_SUCCESS = 0,
 	BE_PY_ERR_APPEND = 6000,
diff -r 559e9710315c -r 789cee7bf738 usr/src/lib/pyzfs/common/ioctl.c
--- a/usr/src/lib/pyzfs/common/ioctl.c	Wed Jan 11 02:37:31 2012 -0800
+++ b/usr/src/lib/pyzfs/common/ioctl.c	Tue Jan 31 04:58:36 2012 -0800
@@ -33,6 +33,10 @@
 #include <libzfs.h>
 #include "zfs_prop.h"
 
+#define B_TRUE _B_TRUE
+#define B_FALSE _B_FALSE
+
+
 static PyObject *ZFSError;
 static int zfsdevfd;
 
diff -r 559e9710315c -r 81d875347eec usr/src/lib/pylibbe/Makefile.com
--- a/usr/src/lib/pylibbe/Makefile.com	Wed Jan 11 02:37:31 2012 -0800
+++ b/usr/src/lib/pylibbe/Makefile.com	Thu Feb 02 03:11:40 2012 -0800
@@ -37,7 +37,7 @@
 PYFILES=	$(PYSRCS) $(PYSRCS:%.py=%.pyc)
 ROOTPYBEFILES=  $(PYFILES:%=$(ROOTLIBDIR)/%)
 
-C99MODE=        $(C99_ENABLE)
+C99MODE=
 
 LIBS =		$(DYNLIB)
 LDLIBS +=	-lbe -lnvpair -lc
diff -r 559e9710315c -r 81d875347eec usr/src/lib/pysolaris/Makefile.com
--- a/usr/src/lib/pysolaris/Makefile.com	Wed Jan 11 02:37:31 2012 -0800
+++ b/usr/src/lib/pysolaris/Makefile.com	Thu Feb 02 03:11:40 2012 -0800
@@ -38,8 +38,8 @@
 ROOTPYSOLFILES= $(PYFILES:%=$(ROOTLIBDIR)/%)
 PYTHON=		$(PYTHON_26)
 
-C99MODE=        -xc99=%all
-C99LMODE=       -Xc99=%all
+C99MODE=
+C99LMODE=
 
 LIBS =		$(DYNLIB)
 LDLIBS +=	-lc -lsec -lidmap -lpython2.6
diff -r 559e9710315c -r 81d875347eec usr/src/lib/pyzfs/Makefile.com
--- a/usr/src/lib/pyzfs/Makefile.com	Wed Jan 11 02:37:31 2012 -0800
+++ b/usr/src/lib/pyzfs/Makefile.com	Thu Feb 02 03:11:40 2012 -0800
@@ -41,8 +41,8 @@
 PYFILES=	$(PYSRCS) $(PYSRCS:%.py=%.pyc)
 ROOTPYZFSFILES= $(PYFILES:%=$(ROOTLIBDIR)/%)
 
-C99MODE=        -xc99=%all
-C99LMODE=       -Xc99=%all
+C99MODE=
+C99LMODE=
 
 LIBS =		$(DYNLIB)
 LDLIBS +=	-lc -lnvpair -lpython2.6 -lzfs
