--- apt-0.8.10.3+squeeze1.orig/apt-pkg/contrib/strutl.cc
+++ apt-0.8.10.3+squeeze1/apt-pkg/contrib/strutl.cc
@@ -69,7 +69,7 @@
   {
      char *outptr = outbuf;
      size_t outsize = bufsize;
-     size_t const err = iconv(cd, &inptr, &insize, &outptr, &outsize);
+     size_t const err = iconv(cd, (const char**)&inptr, &insize, &outptr, &outsize);
      dest->append(outbuf, outptr - outbuf);
      if (err == (size_t)(-1))
      {
