Index: libapache2-mod-encoding-20040616/mod_encoding.c
===================================================================
--- libapache2-mod-encoding-20040616.orig/mod_encoding.c	2011-05-24 01:44:37.760533404 +0400
+++ libapache2-mod-encoding-20040616/mod_encoding.c	2011-05-24 01:44:51.132447464 +0400
@@ -68,6 +68,7 @@
 
 */
 
+#include <regex.h>
 #include <httpd.h>
 #include <http_config.h>
 #include <http_core.h>
