# DP: Issue #8154, fix segfault with os.execlp('true')

Index: b/Modules/posixmodule.c
===================================================================
--- a/Modules/posixmodule.c	2011-08-25 20:59:34.593539534 +0400
+++ b/Modules/posixmodule.c	2011-08-25 20:59:36.383002918 +0400
@@ -2998,6 +2998,11 @@
         PyMem_Free(path);
         return NULL;
     }
+    if (argc < 1) {
+        PyErr_SetString(PyExc_ValueError, "execv() arg 2 must not be empty");
+        PyMem_Free(path);
+        return NULL;
+    }
 
     argvlist = PyMem_NEW(char *, argc+1);
     if (argvlist == NULL) {
