# DP: Proposed fix for issue 9054.

Index: b/Modules/pyexpat.c
===================================================================
--- a/Modules/pyexpat.c	2011-08-25 20:59:10.486329038 +0400
+++ b/Modules/pyexpat.c	2011-08-25 20:59:40.523797494 +0400
@@ -456,6 +456,9 @@
     PyObject *args;
     PyObject *temp;
 
+    if (!have_handler(self, CharacterData))
+        return -1;
+
     args = PyTuple_New(1);
     if (args == NULL)
         return -1;
