Index: b/Lib/pydoc.py
===================================================================
--- a/Lib/pydoc.py	2011-08-25 20:59:21.129377903 +0400
+++ b/Lib/pydoc.py	2011-08-25 20:59:38.787665093 +0400
@@ -1970,10 +1970,14 @@
         if modname[-9:] == '.__init__':
             modname = modname[:-9] + ' (package)'
         print modname, desc and '- ' + desc
+    def onerror(modname):
+        # Ignore non-ImportError exceptions raised whilst trying to
+        # import modules
+        pass
     try: import warnings
     except ImportError: pass
     else: warnings.filterwarnings('ignore') # ignore problems during import
-    ModuleScanner().run(callback, key)
+    ModuleScanner().run(callback, key, onerror=onerror)
 
 # --------------------------------------------------- web browser interface
 
