Index: ntop-40-0/plugins/xmldumpPlugin.c
===================================================================
--- ntop-40-0.orig/plugins/xmldumpPlugin.c	2011-07-07 13:43:02.142633903 +0530
+++ ntop-40-0/plugins/xmldumpPlugin.c	2011-07-07 13:43:06.949467614 +0530
@@ -486,8 +486,10 @@
       return;
     }
 
-    /* Process it */
-    dumpXML(url);
+#ifdef MAKE_WITH_XMLDUMP
+     /* Process it */
+     dumpXML(url);
+#endif
 
 #if defined(PARM_FORK_CHILD_PROCESS) && (!defined(WIN32))
     if(myGlobals.childntoppid != 0)
