Patch from John Davis (upstream) recommended for Debian
AMCK, 2010-03018
diff -ruN slang-2.2.2-old/src/slagetput.inc slang-2.2.2/src/slagetput.inc
--- slang-2.2.2-old/src/slagetput.inc	2009-12-05 19:03:51.000000000 +0000
+++ slang-2.2.2/src/slagetput.inc	2010-03-18 18:33:36.000000000 +0000
@@ -92,7 +92,7 @@
 
 	if ((idx >= 0) 
 	    && ((idx + delta * (jmax-1)) >= 0)
-	    && ((idx + delta * (jmax-1)) <= num_elements))
+	    && ((idx + delta * (jmax-1)) < num_elements))
 	  {
 	     for (j = 0; j < jmax; j++)
 	       {
