Index: pcre3-8.12/configure.ac
===================================================================
--- pcre3-8.12.orig/configure.ac	2011-05-23 14:21:48.231279130 +0400
+++ pcre3-8.12/configure.ac	2011-05-31 00:59:19.453158953 +0400
@@ -14,7 +14,7 @@
 m4_define(pcre_date, [2011-01-15])
 
 # Libtool shared library interface versions (current:revision:age)
-m4_define(libpcre_version, [0:1:0])
+m4_define(libpcre_version, [15:1:12])
 m4_define(libpcreposix_version, [0:0:0])
 m4_define(libpcrecpp_version, [0:0:0])
 
@@ -611,7 +611,7 @@
 # (Note: The libpcre*_version bits are m4 variables, assigned above)
 
 EXTRA_LIBPCRE_LDFLAGS="$EXTRA_LIBPCRE_LDFLAGS \
-                       $NO_UNDEFINED -version-info libpcre_version"
+                       $NO_UNDEFINED -version-info \"libpcre_version\""
 
 EXTRA_LIBPCREPOSIX_LDFLAGS="$EXTRA_LIBPCREPOSIX_LDFLAGS \
                             $NO_UNDEFINED -version-info libpcreposix_version"
