--- a/Makefile.in	2011-03-03 09:00:49.284716850 +0000
+++ b/Makefile.in	2011-03-03 09:16:45.855785181 +0000
@@ -479,6 +479,8 @@
 libsqlite3.la:	$(LIBOBJ)
 	$(LTLINK) -o $@ $(LIBOBJ) $(TLIBS) \
 		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
+	$(LTLINK) -m64 -o amd64/$@ amd64/.libs/$(LIBOBJ) $(TLIBS) \
+		${ALLOWRELEASE} -rpath "$(libdir)/amd64" -version-info "8:6:8"
 	sed -i "/dependency_libs/s/'.*'/''/" $@
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
@@ -516,6 +518,7 @@
 #
 sqlite3.lo:	sqlite3.c
 	$(LTCOMPILE) $(TEMP_STORE) -c sqlite3.c
+	$(LTCOMPILE) -m64 $(TEMP_STORE) -c sqlite3.c -o amd64/.libs/sqlite3.o
 
 # Rules to build the LEMON compiler generator
 #
@@ -888,6 +891,8 @@
 lib_install:	libsqlite3.la
 	$(INSTALL) -d $(DESTDIR)$(libdir)
 	$(LTINSTALL) libsqlite3.la $(DESTDIR)$(libdir)
+	$(INSTALL) -d $(DESTDIR)$(libdir)/amd64
+	$(LTINSTALL) amd64/libsqlite3.la $(DESTDIR)$(libdir)/amd64
 	
 install:	sqlite3$(BEXE) lib_install sqlite3.h sqlite3.pc ${HAVE_TCL:1=tcl_install}
 	$(INSTALL) -d $(DESTDIR)$(bindir)
@@ -918,6 +923,7 @@
 	rm -f testfixture$(TEXE) test.db
 	rm -f sqlite3.dll sqlite3.lib sqlite3.def
 	rm -f sqlite3.c
+	rm -rf amd64
 
 distclean:	clean
 	rm -f config.log config.status libtool Makefile sqlite3.pc
