--- apt-0.8.10.3+squeeze1.orig/cmdline/apt-key
+++ apt-0.8.10.3+squeeze1/cmdline/apt-key
@@ -13,8 +13,8 @@
 #MASTER_KEYRING=/usr/share/keyrings/debian-master-keyring.gpg
 #ARCHIVE_KEYRING_URI=http://ftp.debian.org/debian/debian-archive-keyring.gpg
 
-ARCHIVE_KEYRING=/usr/share/keyrings/debian-archive-keyring.gpg
-REMOVED_KEYS=/usr/share/keyrings/debian-archive-removed-keys.gpg
+ARCHIVE_KEYRING=/usr/share/keyrings/nexenta-archive-keyring.gpg
+REMOVED_KEYS=/usr/share/keyrings/nexenta-archive-removed-keys.gpg
 
 add_keys_with_verify_against_master_keyring() {
     ADD_KEYRING=$1
