Description: Fix LIBS in rlfe
Author: Andrew Stormont (astormont) <andyjstormont@googlemail.com>

--- readline6-6.1.orig/examples/rlfe/Makefile.in
+++ readline6-6.1/examples/rlfe/Makefile.in
@@ -25,7 +25,7 @@ CFLAGS = @CFLAGS@
 CPPFLAGS = @CPPFLAGS@
 #LDFLAGS = -L$(READLINE_DIR)
 LDFLAGS = @LDFLAGS@
-LIBS = -lreadline -lhistory @LIBS@
+LIBS = ../../libreadline.a -lhistory @LIBS@
 
 CPP=@CPP@
 CPP_DEPEND=$(CC) -MM
