Index: apache2-2.2.19/config.layout
===================================================================
--- apache2-2.2.19.orig/config.layout	2004-11-21 21:50:36.000000000 +0300
+++ apache2-2.2.19/config.layout	2011-06-15 11:56:13.615850422 +0400
@@ -322,3 +322,55 @@
     installbuilddir: ${prefix}/etc/apache2/build
     errordir:      ${datadir}/error
 </Layout>
+
+
+# Nexenta layout (like Debian)
+<Layout Nexenta>
+    prefix:
+    exec_prefix:   ${prefix}/usr
+    bindir:        ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/sbin
+    libdir:        ${exec_prefix}/lib
+    libexecdir:    ${exec_prefix}/lib/apache2/modules
+    mandir:        ${exec_prefix}/share/man
+    sysconfdir:    ${prefix}/etc/apache2
+    datadir:       ${exec_prefix}/share/apache2
+    iconsdir:      ${datadir}/icons
+    htdocsdir:     ${prefix}/usr/share/apache2/default-site/htdocs
+    manualdir:     ${htdocsdir}/manual
+    cgidir:        ${prefix}/usr/lib/cgi-bin
+    includedir:    ${exec_prefix}/include/apache2
+    localstatedir: ${prefix}/var/run/apache2
+    runtimedir:    ${prefix}/var/run/apache2
+    logfiledir:    ${prefix}/var/log/apache2
+    proxycachedir: ${prefix}/var/cache/apache2/proxy
+    infodir:       ${exec_prefix}/share/info
+    installbuilddir: ${prefix}/usr/share/apache2/build
+    errordir:      ${datadir}/error
+</Layout>
+
+# CGI dir is the same (some packages put CGI into it)
+# so it is common for i386 and amd64
+<Layout Nexenta-amd64>
+    prefix:
+    exec_prefix:   ${prefix}/usr
+    bindir:        ${exec_prefix}/bin/amd64
+    sbindir:       ${exec_prefix}/sbin/amd64
+    libdir:        ${exec_prefix}/lib/amd64
+    libexecdir:    ${exec_prefix}/lib/apache2/modules/amd64
+    mandir:        ${exec_prefix}/share/man
+    sysconfdir:    ${prefix}/etc/apache2
+    datadir:       ${exec_prefix}/share/apache2
+    iconsdir:      ${datadir}/icons
+    htdocsdir:     ${prefix}/usr/share/apache2/default-site/htdocs
+    manualdir:     ${htdocsdir}/manual
+    cgidir:        ${prefix}/usr/lib/cgi-bin
+    includedir:    ${exec_prefix}/include/apache2
+    localstatedir: ${prefix}/var/run/apache2
+    runtimedir:    ${prefix}/var/run/apache2
+    logfiledir:    ${prefix}/var/log/apache2
+    proxycachedir: ${prefix}/var/cache/apache2/proxy
+    infodir:       ${exec_prefix}/share/info
+    installbuilddir: ${prefix}/usr/share/apache2/build
+    errordir:      ${datadir}/error
+</Layout>
