Index: apache2-2.2.19/configure
===================================================================
--- apache2-2.2.19.orig/configure	2011-05-24 16:06:53.059614581 +0400
+++ apache2-2.2.19/configure	2011-05-24 16:08:03.491516551 +0400
@@ -8273,6 +8273,10 @@
   # Check whether --enable-modules was given.
 if test "${enable_modules+set}" = set; then
   enableval=$enable_modules;
+  if test "$enableval" = "none"; then
+     module_default=no
+     module_selection=none
+  else
     for i in $enableval; do
       if test "$i" = "all" -o "$i" = "most"; then
         module_selection=$i
@@ -8281,7 +8285,7 @@
         eval "enable_$i=yes"
       fi
     done
-
+  fi
 fi
 
 
