Index: apache2-2.2.19/config.layout
===================================================================
--- apache2-2.2.19.orig/config.layout	2011-05-24 16:06:23.230836186 +0400
+++ apache2-2.2.19/config.layout	2011-05-24 16:06:52.975870900 +0400
@@ -314,12 +314,12 @@
     manualdir:     ${htdocsdir}/manual
     cgidir:        ${prefix}/usr/lib/cgi-bin
     includedir:    ${exec_prefix}/include/apache2
-    localstatedir: ${prefix}/var/run
-    runtimedir:    ${prefix}/var/run
+    localstatedir: ${prefix}/var/lock/apache2
+    runtimedir:    ${prefix}/var/run/apache2
     logfiledir:    ${prefix}/var/log/apache2
     proxycachedir: ${prefix}/var/cache/apache2/proxy
     infodir:       ${exec_prefix}/share/info
-    installbuilddir: ${prefix}/etc/apache2/build
+    installbuilddir: ${prefix}/usr/share/apache2/build
     errordir:      ${datadir}/error
 </Layout>
 
Index: apache2-2.2.19/configure
===================================================================
--- apache2-2.2.19.orig/configure	2011-05-20 21:01:46.000000000 +0400
+++ apache2-2.2.19/configure	2011-05-24 16:06:53.059614581 +0400
@@ -22269,17 +22269,17 @@
 
 
 cat >>confdefs.h <<_ACEOF
-#define HTTPD_ROOT "${ap_prefix}"
+#define HTTPD_ROOT "/etc/apache2"
 _ACEOF
 
 
 cat >>confdefs.h <<_ACEOF
-#define SERVER_CONFIG_FILE "${rel_sysconfdir}/${progname}.conf"
+#define SERVER_CONFIG_FILE "${progname}.conf"
 _ACEOF
 
 
 cat >>confdefs.h <<_ACEOF
-#define AP_TYPES_CONFIG_FILE "${rel_sysconfdir}/mime.types"
+#define AP_TYPES_CONFIG_FILE "mime.types"
 _ACEOF
 
 
Index: apache2-2.2.19/configure.in
===================================================================
--- apache2-2.2.19.orig/configure.in	2011-04-16 23:09:47.000000000 +0400
+++ apache2-2.2.19/configure.in	2011-05-24 16:06:53.072643267 +0400
@@ -665,11 +665,11 @@
 echo $MODLIST | $AWK -f $srcdir/build/build-modules-c.awk > modules.c
 
 APR_EXPAND_VAR(ap_prefix, $prefix)
-AC_DEFINE_UNQUOTED(HTTPD_ROOT, "${ap_prefix}",
+AC_DEFINE_UNQUOTED(HTTPD_ROOT, "/etc/apache2",
 	[Root directory of the Apache install area])
-AC_DEFINE_UNQUOTED(SERVER_CONFIG_FILE, "${rel_sysconfdir}/${progname}.conf",
+AC_DEFINE_UNQUOTED(SERVER_CONFIG_FILE, "${progname}.conf",
 	[Location of the config file, relative to the Apache root directory])
-AC_DEFINE_UNQUOTED(AP_TYPES_CONFIG_FILE, "${rel_sysconfdir}/mime.types",
+AC_DEFINE_UNQUOTED(AP_TYPES_CONFIG_FILE, "mime.types",
 	[Location of the MIME types config file, relative to the Apache root directory])
 AC_DEFINE_UNQUOTED(APACHE_MPM_DIR, "$MPM_DIR",
 	[Location of the source for the current MPM])
Index: apache2-2.2.19/include/ap_config_layout.h.in
===================================================================
--- apache2-2.2.19.orig/include/ap_config_layout.h.in	2006-07-12 07:38:44.000000000 +0400
+++ apache2-2.2.19/include/ap_config_layout.h.in	2011-05-24 16:06:53.102860026 +0400
@@ -60,5 +60,6 @@
 #define DEFAULT_REL_LOGFILEDIR "@rel_logfiledir@"
 #define DEFAULT_EXP_PROXYCACHEDIR "@exp_proxycachedir@"
 #define DEFAULT_REL_PROXYCACHEDIR "@rel_proxycachedir@"
+#define DEFAULT_PIDLOG "/var/run/apache2.pid"
 
 #endif /* AP_CONFIG_LAYOUT_H */
