Index: apache2-2.2.19/support/envvars-std.in
===================================================================
--- apache2-2.2.19.orig/support/envvars-std.in	2006-07-12 07:38:44.000000000 +0400
+++ apache2-2.2.19/support/envvars-std.in	2011-05-24 16:07:27.700131685 +0400
@@ -18,7 +18,5 @@
 #
 # This file is generated from envvars-std.in
 #
-@SHLIBPATH_VAR@="@exp_libdir@:$@SHLIBPATH_VAR@"
-export @SHLIBPATH_VAR@
 #
 @OS_SPECIFIC_VARS@
