diff -ur orig/dpkg-1.15.8.5stormos0/configure dpkg-1.15.8.5stormos0/configure
--- orig/dpkg-1.15.8.5stormos0/configure	2010-10-04 03:10:09.000000000 +0530
+++ dpkg-1.15.8.5stormos0/configure	2010-11-03 12:25:21.717969135 +0530
@@ -10528,7 +10528,7 @@
 fi
 
 
-WFLAGS="-Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers \
+WFLAGS="-Wall -Wextra -Wno-unused-parameter \
 	 -Wmissing-declarations -Wmissing-format-attribute \
 	 -Winit-self -Wwrite-strings -Wcast-align -Wshadow"
 WCFLAGS="-Wdeclaration-after-statement -Wnested-externs -Wbad-function-cast \
