diff -ur orig/dpkg-1.15.8.5stormos0/dselect/baselist.cc dpkg-1.15.8.5stormos0/dselect/baselist.cc
--- orig/dpkg-1.15.8.5stormos0/dselect/baselist.cc	2010-09-14 05:12:33.000000000 +0530
+++ dpkg-1.15.8.5stormos0/dselect/baselist.cc	2010-11-03 12:45:46.267589153 +0530
@@ -55,7 +55,7 @@
   p->enddisplay();
   endwin(); initscr();
   if (ioctl(fileno(stdout), TIOCGWINSZ, &size) != 0) ohshite(_("ioctl(TIOCGWINSZ) failed"));
-  resizeterm(size.ws_row, size.ws_col); wrefresh(curscr);
+  //resizeterm(size.ws_row, size.ws_col); wrefresh(curscr);
   p->startdisplay();
   if (doupdate() == ERR) ohshite(_("doupdate in SIGWINCH handler failed"));
 }
