--- a/cheetah/Compiler.py	2010-02-08 05:17:23.000000000 +0100
+++ b/cheetah/Compiler.py	2010-06-08 06:50:28.000000000 +0200
@@ -1474,8 +1474,12 @@
         return '\n\n'.join(methodDefs)
 
     def attributes(self):
-        attribs = [self.setting('indentationStep') + str(attrib)
-                      for attrib in self._generatedAttribs ]
+        try:
+            attribs = [self.setting('indentationStep') + str(attrib)
+                          for attrib in self._generatedAttribs ]
+        except UnicodeEncodeError:
+            attribs = [self.setting('indentationStep') + unicode(attrib)
+                          for attrib in self._generatedAttribs ]
         return '\n\n'.join(attribs)
   
 class AutoClassCompiler(ClassCompiler):
