Index: apache2-2.2.19/modules/mappers/mod_rewrite.c
===================================================================
--- apache2-2.2.19.orig/modules/mappers/mod_rewrite.c	2011-03-07 23:39:54.000000000 +0300
+++ apache2-2.2.19/modules/mappers/mod_rewrite.c	2011-05-24 16:08:23.144730975 +0400
@@ -840,6 +840,23 @@
 }
 
 /*
+ * Get the current server name from the request for the purposes
+ * of using in a URL.  If the server name is an IPv6 literal
+ * address, it will be returned in URL format (e.g., "[fe80::1]").
+ */
+static const char *get_server_name_for_url(request_rec *r)
+{
+    const char *plain_server_name = ap_get_server_name(r);
+
+#if APR_HAVE_IPV6
+    if (ap_strchr_c(plain_server_name, ':')) { /* IPv6 literal? */
+        return apr_psprintf(r->pool, "[%s]", plain_server_name);
+    }
+#endif
+    return plain_server_name;
+}
+
+/*
  * add 'http[s]://ourhost[:ourport]/' to URI
  * if URI is still not fully qualified
  */
@@ -853,7 +870,7 @@
         char *thisport;
         int port;
 
-        thisserver = ap_get_server_name(r);
+        thisserver = get_server_name_for_url(r);
         port = ap_get_server_port(r);
         thisport = ap_is_default_port(port, r)
                    ? ""
@@ -4290,7 +4307,7 @@
      */
 
     /* add the canonical URI of this URL */
-    thisserver = ap_get_server_name(r);
+    thisserver = get_server_name_for_url(r);
     port = ap_get_server_port(r);
     if (ap_is_default_port(port, r)) {
         thisport = "";
