Description: Switch from using distutils to setuptools.
Author: Daniel Watkins <daniel@daniel-watkins.co.uk>
Forwarded: not-needed
Last-Update: 2010-01-31

--- configobj_4.5.2.orig/setup.py
+++ configob_4.5.2/setup.py
@@ -9,7 +9,7 @@
 # http://www.voidspace.org.uk/python/license.shtml
 
 import sys
-from distutils.core import setup
+from setuptools import setup
 from configobj import __version__ as VERSION
 
 NAME = 'configobj'
