## disabledby-pythonindent.diff by James Vega <jamessan@debian.org>

diff -urNad vim-scripts-7.1.5~/indent/python.vim vim-scripts-7.1.5/indent/python.vim
--- vim-scripts-7.1.5~/indent/python.vim	2007-12-12 09:15:07.000000000 -0500
+++ vim-scripts-7.1.5/indent/python.vim	2007-12-12 10:37:07.000000000 -0500
@@ -5,7 +5,7 @@
 " Last Change:      2004 Jun 07
 
 " Only load this indent file when no other was loaded.
-if exists("b:did_indent")
+if exists("b:did_indent") || exists("loaded_python_indent")
     finish
 endif
 let b:did_indent = 1
