## disabledby-closetag.diff by Stefano Zacchiroli <zack@debian.org>

Index: vim-scripts/macros/closetag.vim
===================================================================
--- vim-scripts.orig/macros/closetag.vim
+++ vim-scripts/macros/closetag.vim
@@ -121,6 +121,12 @@
 " User configurable settings
 "------------------------------------------------------------------------------
 
+" Has this already been loaded?
+if exists("loaded_closetag")
+    finish
+endif
+let loaded_closetag=1
+
 " if html, don't close certain tags.  Works best if ignorecase is set.
 " otherwise, capitalize these elements according to your html editing style
 if !exists("b:unaryTagsStack") || exists("b:closetag_html_style")
@@ -131,12 +137,6 @@
     endif
 endif
 
-" Has this already been loaded?
-if exists("loaded_closetag")
-    finish
-endif
-let loaded_closetag=1
-
 " set up mappings for tag closing
 inoremap <C-_> <C-R>=GetCloseTag()<CR>
 map <C-_> a<C-_><ESC>
