## disabledby-xml.diff by Stefano Zacchiroli <zack@debian.org>

Index: vim-scripts/ftplugin/xml.vim
===================================================================
--- vim-scripts.orig/ftplugin/xml.vim
+++ vim-scripts/ftplugin/xml.vim
@@ -44,7 +44,7 @@
 "==============================================================================
 
 " Only do this when not done yet for this buffer
-if exists("b:did_ftplugin")
+if exists("b:did_ftplugin") || exists("loaded_xml_ftplugin")
   finish
 endif
 " sboles, init these variables so vim doesn't complain on wrap cancel
