diff -u reprepro-4.12.3.orig/dirs.c reprepro-4.12.3/dirs.c
--- reprepro-4.12.3.orig/dirs.c	2012-06-08 02:27:13.000000000 +0530
+++ reprepro-4.12.3/dirs.c	2012-09-07 15:59:46.300233540 +0530
@@ -166,7 +166,8 @@
 					this);
 		} else {
 			int e = errno;
-			if (e != ENOTEMPTY) {
+			int en = errno;
+			if (e != ENOTEMPTY && en != EEXIST) {
 				fprintf(stderr,
 "Error removing directory \"%s\": %s(%d)\n",
 						this, strerror(e), e);
diff -u reprepro-4.12.3.orig/pool.c reprepro-4.12.3/pool.c
--- reprepro-4.12.3.orig/pool.c	2012-06-08 02:27:13.000000000 +0530
+++ reprepro-4.12.3/pool.c	2012-09-07 15:58:14.101169754 +0530
@@ -474,7 +474,7 @@
 				}
 			} else {
 				en = errno;
-				if (en != ENOTEMPTY) {
+				if (en != ENOTEMPTY && en != EEXIST) {
 					//TODO: check here if only some
 					//other error was first and it
 					//is not empty so we do not have
