diff -dubr reprepro-4.12.3/extractcontrol.c reprepro-4.12.3.gtar/extractcontrol.c
--- reprepro-4.12.3/extractcontrol.c	2012-06-08 00:57:13.000000000 +0400
+++ reprepro-4.12.3.gtar/extractcontrol.c	2012-10-05 15:38:39.616187754 +0400
@@ -111,11 +111,11 @@
 		close(pipe1[0]); close(pipe1[1]);
 		close(pipe2[0]); close(pipe2[1]);
 		//TODO without explicit path
-		execl("/bin/tar", "tar", "-xOzf", "-",
+		execl("/bin/gtar", "gtar", "-xOzf", "-",
 				brokentar?"control":"./control",
 				ENDOFARGUMENTS);
 		e = errno;
-		fprintf(stderr, "tar call failed with error %d: %s\n",
+		fprintf(stderr, "gtar call failed with error %d: %s\n",
 				e, strerror(e));
 		exit(254);
 
@@ -133,8 +133,8 @@
 
 		r = readtextfilefd(pipe2[0],
 				brokentar?
-"output from ar p <debfile> control.tar.gz | tar -XOzf - control":
-"output from ar p <debfile> control.tar.gz | tar -XOzf - ./control",
+"output from ar p <debfile> control.tar.gz | gtar -XOzf - control":
+"output from ar p <debfile> control.tar.gz | gtar -XOzf - ./control",
 				&controlchunk, &controllen);
 		if (RET_IS_OK(r)) {
 			len = chunk_extract(controlchunk,
@@ -145,7 +145,7 @@
 			if (*afterchanges != '\0') {
 				fprintf(stderr,
 "Unexpected emtpy line in control information within '%s'\n"
-"(obtained via 'ar p %s control.tar.gz | tar -XOzf - %scontrol')\n",
+"(obtained via 'ar p %s control.tar.gz | gtar -XOzf - %scontrol')\n",
 						debfile, debfile,
 						brokentar?"":"./");
 				free(controlchunk);
