Description: Return failure if doctests fail
Author: Stefano Rivera <stefanor@debian.org>
Forwarded: http://code.google.com/p/configobj/issues/detail?id=30
Last-Update: 2011-02-05
--- a/tests/test_configobj.py
+++ b/tests/test_configobj.py
@@ -2215,7 +2215,10 @@
     doctest.testmod(m, globs=globs)
     
     import configobj
-    doctest.testmod(configobj, globs=globs)
+    failures, tests = doctest.testmod(configobj, globs=globs)
+    print "Ran %i tests. Failures: %i" % (tests, failures)
+    if failures:
+        sys.exit(1)
 
 
-# Man alive I prefer unittest ;-)
\ No newline at end of file
+# Man alive I prefer unittest ;-)
