Index: python-cherrypy-2.3.0/cherrypy/_cphttptools.py
===================================================================
--- python-cherrypy-2.3.0.orig/cherrypy/_cphttptools.py	2011-07-19 11:57:15.016867015 +0400
+++ python-cherrypy-2.3.0/cherrypy/_cphttptools.py	2011-07-19 11:57:25.485427266 +0400
@@ -200,7 +200,16 @@
             # Handle cookies differently because on Konqueror, multiple
             # cookies come on different lines with the same key
             if name.title() == 'Cookie':
-                self.simple_cookie.load(value)
+                # Cookies with a colon (":") are invalid according to rfc2965
+                # and rfc2068. However if the browser send such a cookie, we
+                # want to ignore it and continue instead of returning an
+                # "500 Internal Server Error" error. More infos on:
+                # http://www.cherrypy.org/ticket/868
+                try:
+                    self.simple_cookie.load(value)
+                except Cookie.CookieError, e:
+                    cherrypy.log("Unable to load user's cookie. Cookie ignored.")
+
         
         # Save original values (in case they get modified by filters)
         # This feature is deprecated in 2.2 and will be removed in 2.3.
