Index: python-cherrypy-2.3.0/cherrypy/lib/profiler.py
===================================================================
--- python-cherrypy-2.3.0.orig/cherrypy/lib/profiler.py	2011-07-19 11:56:19.081417253 +0400
+++ python-cherrypy-2.3.0/cherrypy/lib/profiler.py	2011-07-19 11:56:31.464825787 +0400
@@ -46,12 +46,12 @@
 except ImportError:
     profile = None
     pstats = None
-    import warnings
-    msg = ("Your installation of Python doesn't have a profile module. "
-           "If you're on Debian, you can apt-get python2.4-profiler from "
-           "non-free in a separate step. See http://www.cherrypy.org/wiki/"
-           "ProfilingOnDebian for details.")
-    warnings.warn(msg)
+    #import warnings
+    #msg = ("Your installation of Python doesn't have a profile module. "
+    #       "If you're on Debian, you can apt-get python2.4-profiler from "
+    #       "non-free in a separate step. See http://www.cherrypy.org/wiki/"
+    #       "ProfilingOnDebian for details.")
+    #warnings.warn(msg)
 
 import os, os.path
 import sys
