Index: lib/aliases.gperf
===================================================================
--- lib/aliases.gperf	(.../pool/current)	(revision 20162)
+++ lib/aliases.gperf	(.../trunk)	(revision 20162)
@@ -11,6 +11,7 @@
 %%
 US-ASCII, ei_ascii
 ASCII, ei_ascii
+646, ei_ascii
 ISO646-US, ei_ascii
 ISO_646.IRV:1991, ei_ascii
 ISO-IR-6, ei_ascii
--- src/Makefile.in.orig	2006-10-12 01:11:19.843813000 -0700
+++ src/Makefile.in	2006-10-12 01:11:35.741059000 -0700
@@ -96,7 +96,7 @@
 	if [ ! -d $(DESTDIR)$(bindir) ] ; then $(mkinstalldirs) $(DESTDIR)$(bindir) ; fi
 	case "@host_os@" in \
 	  hpux*) $(CC) $(LDFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a -L$(DESTDIR)$(libdir) -liconv @LIBINTL@ `if test -n '$(DESTDIR)'; then echo " -Wl,+b -Wl,$(libdir)"; fi` -o iconv;; \
-	  *) $(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a $(DESTDIR)$(libdir)/libiconv.la @LTLIBINTL@ -o iconv;; \
+	  *) $(LIBTOOL_LINK) $(CC) $(LDFLAGS) $(iconv_LDFLAGS) iconv.@OBJEXT@ ../srclib/libicrt.a ../lib/libiconv.la @LTLIBINTL@ -o iconv;; \
 	esac
 	$(INSTALL_PROGRAM_ENV) $(LIBTOOL_INSTALL) $(INSTALL_PROGRAM) iconv $(DESTDIR)$(bindir)/iconv
 
--- lib/encodings.def.orig	2006-10-12 08:27:49.541183000 -0700
+++ lib/encodings.def	2006-10-12 08:28:18.388860000 -0700
@@ -47,6 +47,7 @@
               "IBM367",                 /* IANA */
               "US",                     /* IANA */
               "csASCII",                /* IANA */
+	      "646",			/* Solaris */
             /*"ISO646.1991-IRV",           X11R6.4 */
             ),
             ascii,
