Index: ncurses-5.7+20100313/c++/cursesmain.cc
===================================================================
--- ncurses-5.7+20100313.orig/c++/cursesmain.cc	2011-05-18 13:41:36.248642130 +0400
+++ ncurses-5.7+20100313/c++/cursesmain.cc	2011-05-18 13:41:48.016222477 +0400
@@ -34,12 +34,12 @@
 #include "internal.h"
 #include "cursesapp.h"
 
-#if CPP_HAS_TRY_CATCH && HAVE_IOSTREAM
-#include <iostream>
-#else
+//#if CPP_HAS_TRY_CATCH && HAVE_IOSTREAM
+//#include <iostream>
+//#else
 #undef CPP_HAS_TRY_CATCH
 #define CPP_HAS_TRY_CATCH 0
-#endif
+//#endif
 
 MODULE_ID("$Id: cursesmain.cc,v 1.14 2007/04/07 17:10:11 tom Exp $")
 
