--- apt-0.8.15.10+1.orig/cmdline/apt-key	2012-10-02 03:21:53.601644475 -0700
+++ apt-0.8.15.10+1/cmdline/apt-key	2012-10-02 03:27:55.540860070 -0700
@@ -1,17 +1,19 @@
-#!/bin/sh
+#!/bin/bash
 
 set -e
 unset GREP_OPTIONS
 
+GPGBIN="gpg2"
+
 # We don't use a secret keyring, of course, but gpg panics and
 # implodes if there isn't one available
-SECRETKEYRING="$(mktemp)"
+SECRETKEYRING="/tmp/tempkey.$$"
 trap "rm -f '${SECRETKEYRING}'" 0 HUP INT QUIT ILL ABRT FPE SEGV PIPE TERM
-GPG_CMD="gpg --ignore-time-conflict --no-options --no-default-keyring --secret-keyring ${SECRETKEYRING}"
+GPG_CMD="$GPGBIN --ignore-time-conflict --no-options --no-default-keyring --secret-keyring ${SECRETKEYRING}"
 
 if [ "$(id -u)" -eq 0 ]; then
 	# we could use a tmpfile here too, but creation of this tends to be time-consuming
-	GPG_CMD="$GPG_CMD --trustdb-name /etc/apt/trustdb.gpg"
+	GPG_CMD="$GPG_CMD --trustdb-name ${BASEDIR}/etc/apt/trustdb.gpg"
 fi
 
 GPG="$GPG_CMD"
@@ -79,15 +81,15 @@
 	echo >&2 "ERROR: an installed wget is required for a network-based update"
 	exit 1
     fi
-    if [ ! -d /var/lib/apt/keyrings ]; then
-	mkdir -p /var/lib/apt/keyrings
+    if [ ! -d ${BASEDIR}/var/lib/apt/keyrings ]; then
+	mkdir -p ${BASEDIR}/var/lib/apt/keyrings
     fi
-    keyring=/var/lib/apt/keyrings/$(basename $ARCHIVE_KEYRING)
+    keyring=${BASEDIR}/var/lib/apt/keyrings/$(basename $ARCHIVE_KEYRING)
     old_mtime=0
     if [ -e $keyring ]; then
 	old_mtime=$(stat -c %Y $keyring)
     fi
-    (cd  /var/lib/apt/keyrings; wget -q -N $ARCHIVE_KEYRING_URI)
+    (cd  ${BASEDIR}/var/lib/apt/keyrings; wget -q -N $ARCHIVE_KEYRING_URI)
     if [ ! -e $keyring ]; then
 	return
     fi
@@ -101,7 +103,7 @@
 update() {
     if [ ! -f $ARCHIVE_KEYRING ]; then
 	echo >&2 "ERROR: Can't find the archive-keyring"
-	echo >&2 "Is the debian-archive-keyring package installed?"
+	echo >&2 "Is the archive-keyring package installed?"
 	exit 1
     fi
     requires_root
@@ -165,15 +167,17 @@
 	TRUSTEDFILE="/etc/apt/trusted.gpg"
 	eval $(apt-config shell TRUSTEDFILE Apt::GPGV::TrustedKeyring)
 	eval $(apt-config shell TRUSTEDFILE Dir::Etc::Trusted/f)
+	TRUSTEDFILE="${BASEDIR}$TRUSTEDFILE"
 	if [ -r "$TRUSTEDFILE" ]; then
 		GPG="$GPG --keyring $TRUSTEDFILE"
 	fi
 	GPG="$GPG --primary-keyring $TRUSTEDFILE"
 	TRUSTEDPARTS="/etc/apt/trusted.gpg.d"
 	eval $(apt-config shell TRUSTEDPARTS Dir::Etc::TrustedParts/d)
+	TRUSTEDPARTS="${BASEDIR}$TRUSTEDPARTS"
 	if [ -d "$TRUSTEDPARTS" ]; then
 		#echo "parts active"
-		for trusted in $(run-parts --list $TRUSTEDPARTS --regex '^.*\.gpg$'); do
+		for trusted in $(find $TRUSTEDPARTS -name '^.*\.gpg$'); do
 			#echo "part -> $trusted"
 			GPG="$GPG --keyring $trusted"
 		done
@@ -188,7 +192,7 @@
 fi
 shift
 
-if [ "$command" != "help" ] && ! which gpg >/dev/null 2>&1; then
+if [ "$command" != "help" ] && ! which $GPGBIN >/dev/null 2>&1; then
     echo >&2 "Warning: gnupg does not seem to be installed."
     echo >&2 "Warning: apt-key requires gnupg for most operations."
     echo >&2
