diff -ru apt-0.8.10.3+squeeze1/doc/manpage-style.xsl apt-0.8.10.3+squeeze1.patched/doc/manpage-style.xsl
--- apt-0.8.10.3+squeeze1/doc/manpage-style.xsl	2011-04-15 11:30:33.000000000 +0400
+++ apt-0.8.10.3+squeeze1.patched/doc/manpage-style.xsl	2011-11-30 22:26:43.820612650 +0300
@@ -2,7 +2,7 @@
  xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
  version="1.0">
 
-<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/manpages/docbook.xsl" />
+<xsl:import href="/usr/share/sgml/docbook/xsl-stylesheets-1.69.1-5.1/manpages/docbook.xsl" />
 
 <xsl:param name="man.output.encoding" select="'UTF-8'" />
 
