diff -urN apt-0.8.15.10+1.old/cmdline/apt-get.cc apt-0.8.15.10+1/cmdline/apt-get.cc
--- apt-0.8.15.10+1.old/cmdline/apt-get.cc	2012-01-11 05:42:59.000000000 -0800
+++ apt-0.8.15.10+1/cmdline/apt-get.cc	2014-01-27 22:43:18.469014838 -0800
@@ -3344,6 +3346,8 @@
 									/*}}}*/
 int main(int argc,const char *argv[])					/*{{{*/
 {
+   char *path;
+   char newpath[512];
    CommandLine::Args Args[] = {
       {'h',"help","help",0},
       {'v',"version","version",0},
@@ -3417,6 +3423,17 @@
    setlocale(LC_ALL,"");
    textdomain(PACKAGE);
 
+   // Set gnu PATH
+   path = getenv ("PATH");
+   if (path != NULL) {
+               strlcpy (newpath, "/usr/gnu/bin:", sizeof (newpath));
+               strlcat (newpath, path, sizeof (newpath));
+   } else {
+               strlcpy (newpath, "/usr/gnu/bin:", sizeof (newpath));
+               strlcat (newpath, "/usr/bin:/usr/sbin:/sbin", sizeof (newpath));
+   }
+   setenv ("PATH", newpath, 1);
+
    // Parse the command line and initialize the package library
    CommandLine CmdL(Args,_config);
    if (pkgInitConfig(*_config) == false ||
