Index: debhelper-8.1.6/dh
===================================================================
--- debhelper-8.1.6.orig/dh	2011-05-16 14:21:21.000000000 +0400
+++ debhelper-8.1.6/dh	2011-05-27 17:10:54.838279626 +0400
@@ -350,6 +350,7 @@
 	dh_installifupdown
 	dh_installinfo
 	dh_installinit
+	dh_installsvc
 	dh_installmenu
 	dh_installmime
 	dh_installmodules
@@ -370,6 +371,7 @@
 	dh_usrlocal
 
 	dh_link
+	dh_isaexec
 	dh_compress
 	dh_fixperms
 };
