Index: debhelper/dh
===================================================================
--- debhelper.orig/dh	2011-07-05 14:37:05.734254125 +0400
+++ debhelper/dh	2011-07-05 14:39:09.945087240 +0400
@@ -350,6 +350,7 @@
 	dh_installifupdown
 	dh_installinfo
 	dh_installinit
+	dh_installsvc
 	dh_installmenu
 	dh_installmime
 	dh_installmodules
@@ -368,8 +369,10 @@
 	dh_icons
 	dh_perl
 	dh_usrlocal
+	dh_obfuscate
 
 	dh_link
+	dh_isaexec
 	dh_compress
 	dh_fixperms
 };
