# HG changeset patch
# Parent 568e6dd54ce86511bd0adefc41b4e3ea55f98b1b

diff -r 568e6dd54ce8 usr/src/pkg/Makefile
--- a/usr/src/pkg/Makefile	Thu Mar 22 02:15:13 2012 +0400
+++ b/usr/src/pkg/Makefile	Fri Mar 23 16:32:13 2012 +0400
@@ -26,6 +26,8 @@
 include $(SRC)/Makefile.master
 include $(SRC)/Makefile.buildnum
 
+PKGS_TYPE= ips
+
 #
 # Make sure we're getting a consistent execution environment for the
 # embedded scripts.
@@ -96,6 +98,8 @@
 TOOLSROOT=	$(TOOLSROOT.cmd:sh)
 PKGDEST.cmd=	print $(PKGARCHIVE) | sed -e s:/$(MACH)/:/$(PKGMACH)/:
 PKGDEST=	$(PKGDEST.cmd:sh)
+PKGDESTAPT.cmd=	print $(PKGARCHIVEAPT) | sed -e s:/$(MACH)/:/$(PKGMACH)/:
+PKGDESTAPT=	$(PKGDESTAPT.cmd:sh)
 
 EXCEPTIONS= packaging
 
@@ -120,6 +124,7 @@
 # Nothing underneath $(PDIR) should ever be managed by SCM.
 #
 PDIR= packages.$(PKGMACH)
+PDIRDEB= $(CODEMGR_WS)/packages/$(PKGMACH)/debs
 
 #
 # The tools proto must be specified for dependency generation.
@@ -137,6 +142,9 @@
 PKGVERS_BUILTON= $(RELEASE)
 PKGVERS_BRANCH= 0.$(ONNV_BUILDNUM)
 PKGVERS= $(PKGVERS_COMPONENT),$(PKGVERS_BUILTON)-$(PKGVERS_BRANCH)
+DEB_VERSION= 1.0.$(ONNV_BUILDNUM)-2
+DEB_MAINTAINER= Maintainer <maintainer@illumos.org>
+DPKG_BUILD_FLAGS= -d -b -uc
 
 #
 # The ARCH32 and ARCH64 macros are used in the manifests to express
@@ -218,6 +226,11 @@
 DEP_PKGS= $(PKGS:%=$(PDIR)/%.dep)
 PROC_PKGS= $(PKGS:%=$(PDIR)/%.mog)
 
+DEB_PKGS= $(PKGS:%=$(PDIRDEB)/%.deb.stamp)
+DEB_PKGS_CHANGES.cmd= print $(PDIRDEB)/*.changes
+DEB_PKGS_CHANGES= $(DEB_PKGS_CHANGES.cmd:sh)
+DEB_PKGS_APT= $(DEB_PKGS_CHANGES:%_$(DEB_VERSION)_solaris-$(MACH).changes=%.apt)
+
 #
 # Track the synthetic manifests separately so we can properly express
 # build rules and dependencies.  The synthetic and real packages use
@@ -270,7 +283,9 @@
 
 .KEEP_STATE:
 
-.PARALLEL: $(PKGS) $(PROC_PKGS) $(DEP_PKGS) \
+.NO_PARALLEL: debapt $(DEB_PKGS_APT) debaptconf debaptindex
+
+.PARALLEL: $(PKGS) $(PROC_PKGS) $(DEB_PKGS) $(DEP_PKGS) \
 	$(PROC_SYNTH_PKGS) $(DEP_SYNTH_PKGS) $(PUB_PKGS)
 
 #
@@ -356,6 +371,14 @@
 	@print "Creating $(@)"
 	$(PKGDEBUG)$(INS.dir)
 
+$(PDIRDEB):
+	@print "Creating $(@)"
+	$(PKGDEBUG)$(INS.dir)
+
+$(PKGDESTAPT):
+	@print "Creating $(@)"
+	$(PKGDEBUG)$(INS.dir)
+
 #
 # This rule resolves dependencies across all published manifests.
 #
@@ -389,7 +412,16 @@
 	fi
 	$(PKGDEBUG)$(TOUCH) $(@)
 
-install: $(ALL_TARGETS) repository-metadata
+install: install_$(PKGS_TYPE)
+
+install_ips: $(ALL_TARGETS) repository-metadata
+
+install_deb: deb .WAIT debapt
+
+deb:	$(PDIRDEB) .WAIT $(DEB_PKGS)
+
+debapt: debaptindex
+
 
 repository-metadata: publish_pkgs
 	@print "Creating repository metadata"
@@ -426,6 +458,14 @@
 		--set-property publisher.prefix=$(PKGPUBLISHER)
 
 #
+# Initialize the empty on-disk repositories
+#
+$(PKGDESTAPT)/conf/distributions: $(PKGDESTAPT)
+	@print "Configuring APT repo"
+	$(PKGDEBUG)mkdir -p $(PKGDESTAPT)/conf
+	$(PKGDEBUG)$(CP) /opt/onbld/etc/distributions.apt $(PKGDESTAPT)/conf/distributions
+
+#
 # rule to process real manifests
 #
 # To allow redistributability and package status to change, we must
@@ -509,6 +549,48 @@
 	fi
 	$(PKGDEBUG)$(RM) $(@).vars
 
+
+$(PDIRDEB)/%.deb.stamp: $(PDIR)/%.mog
+	$(PKGDEBUG)(sudo  \
+	/opt/onbld/bin/ips2deb.pl --noverbose \
+	-p $< \
+	-d $(PKGROOT) \
+	-d $(TOOLSROOT) \
+	-o "$(PDIRDEB)" \
+	--mfg "$(PDIR)" \
+	--mfe "mog" \
+	--maintainer "$(DEB_MAINTAINER)" \
+	--pv "$(DEB_VERSION)" \
+	--category "illumos" ; \
+	d=`echo $(<F:%.mog=%) | $(SED) -e 's/[\/_]/-/g' | $(NAWK) '{print tolower($1)}'` ; \
+	if [ -d "$(PDIRDEB)/$$d" ]; then \
+	    $(TOUCH) $(PDIRDEB)/$$d.build.log ; \
+	    echo "Building pkg: $$d" ; \
+	    cd "$(PDIRDEB)/$$d" ; \
+	    sudo PATH=/usr/gnu/bin:/usr/bin:$$PATH \
+	    dpkg-buildpackage $(DPKG_BUILD_FLAGS) 2> ../$$d.build.log 1>&2 ; \
+	fi )
+	@$(TOUCH) $@
+
+debaptindex: debapt2
+	@print "Indexing APT repository"
+	$(PKGDEBUG)/usr/bin/reprepro -b $(PKGDESTAPT) export
+	@print "Local APT repo '$(PKGDESTAPT)' is ready to use"
+
+debapt2: $(PKGDESTAPT)/conf/distributions
+	$(PKGDEBUG)for p in $(PDIRDEB)/*.changes; do \
+	    print "Sending $$p to the local APT ..." ; \
+	    /usr/bin/reprepro -b $(PKGDESTAPT) --export=never remove unstable `$(ECHO) $$p | $(SED) -e 's:.*/::' | cut -f1 -d'_'` 2> /dev/null 1>&2; \
+	    /usr/bin/reprepro -b $(PKGDESTAPT) --export=never include unstable $$p 2> /dev/null 1>&2 ; \
+	    $(TOUCH) `$(ECHO) $$p | cut -f1 -d'_'`.apt ; \
+	done
+
+$(PDIRDEB)/%.apt: $(PDIRDEB)/%_$(DEB_VERSION)_solaris-$(MACH).changes
+	$(PKGDEBUG)print "Sending pkg '$(@F:%.apt=%)' to APT ..."; \
+	    /usr/bin/reprepro -b $(PKGDESTAPT) --export=never remove unstable $(@F:%.apt=%) 2> /dev/null 1>&2; \
+	    /usr/bin/reprepro -b $(PKGDESTAPT) --export=never include unstable $(<) 2> /dev/null 1>&2
+	@$(TOUCH) $@
+
 $(PDIR)/%.dep: $(PDIR)/%.mog
 	@print "Generating dependencies for $(<F)"
 	$(PKGDEBUG)$(RM) $(@)
@@ -570,11 +652,21 @@
 	@print "Skipping dependency generation for $(@F:%.dep=%)"
 	$(PKGDEBUG)$(CP) $(@:%.dep=%.mog) $(@)
 
-clean:
+clean: clean_$(PKGS_TYPE)
 
-clobber: clean
+clean_ips:
+
+clean_deb:
+	sudo $(RM) -r $(PDIRDEB)
+
+clobber: clobber_$(PKGS_TYPE)
+
+clobber_ips: clean_ips
 	$(RM) -r $(CLOBBERFILES)
 
+clobber_deb: clean_deb
+	$(RM) -r $(PKGDESTAPT)
+
 #
 # This rule assumes that all links in the $PKGSTAT directories
 # point to valid manifests, and will fail the make run if one
