--- /dev/null				2011-07-27 00:08:35.000000000 +0100
+++ ./data/templates/Nexenta.info.in	2011-07-26 23:18:01.555910895 +0100
@@ -0,0 +1,48 @@
+# NCP 4.0 - based on squeeze
+Suite: siddy-stable
+RepositoryType: deb
+BaseURI: http://apt.nexenta.org/wip
+MatchName: stable
+MirrorsFile: Nexenta.mirrors
+Description: NCP 4.0 - "siddy-stable"
+Component: main
+CompDescription: Main supported software
+CompDescriptionLong: Core software
+Component: contrib
+CompDescription: Community supported software
+CompDescriptionLong: Extra software
+Component: non-free
+CompDescription: Non-free software
+CompDescriptionLong: Non-free software
+
+Suite: siddy-testing
+RepositoryType: deb
+BaseURI: http://apt.nexenta.org/wip
+MatchName: testing
+MirrorsFile: Nexenta.mirrors
+Description: NCP 4.0 - "siddy-testing"
+Component: main
+CompDescription: Main supported software
+CompDescriptionLong: Core software
+Component: contrib
+CompDescription: Community supported software
+CompDescriptionLong: Extra software
+Component: non-free
+CompDescription: Non-free software
+CompDescriptionLong: Non-free software
+
+Suite: siddy-unstable
+RepositoryType: deb
+BaseURI: http://apt.nexenta.org/wip
+MatchName: unstable
+MirrorsFile: Nexenta.mirrors
+Description: NCP 4.0 - "siddy-unstable"
+Component: main
+CompDescription: Main supported software
+CompDescriptionLong: Core software
+Component: contrib
+CompDescription: Community supported software
+CompDescriptionLong: Extra software
+Component: non-free
+CompDescription: Non-free software
+CompDescriptionLong: Non-free software
--- /dev/null				2011-07-27 00:08:35.000000000 +0100
+++ ./data/templates/Nexenta.mirrors	2011-07-26 21:25:14.801231383 +0100
@@ -0,0 +1,2 @@
+#LOC:US
+http://apt.nexenta.org/wip/
