diff -urN net-snmp-5.4.1.orig/agent/agent_registry.c net-snmp-5.4.1/agent/agent_registry.c
--- net-snmp-5.4.1.orig/agent/agent_registry.c	2007-07-16 16:59:44.000000000 -0700
+++ net-snmp-5.4.1/agent/agent_registry.c	2013-08-21 08:43:51.575564973 -0700
@@ -532,7 +532,8 @@
 	
 	    if (next && (next->namelen  == new_sub->namelen) &&
 		(next->priority == new_sub->priority)) {
-                netsnmp_assert(!"registration != duplicate"); /* always false */
+		if (new_sub->namelen != 1) /* ignore root OID dups */
+		    netsnmp_assert(!"registration != duplicate"); /* always false */
 		return MIB_DUPLICATE_REGISTRATION;
 	    }
 
