diff -urN net-snmp-5.5.upstream/include/net-snmp/net-snmp-config.h.in net-snmp-5.5/include/net-snmp/net-snmp-config.h.in
--- net-snmp-5.5.upstream/include/net-snmp/net-snmp-config.h.in	2009-08-04 00:57:18.000000000 +0530
+++ net-snmp-5.5/include/net-snmp/net-snmp-config.h.in	2010-03-16 06:18:40.000000000 +0530
@@ -24,6 +24,13 @@
  */
 #ifndef NETSNMP_NO_AUTOCONF_DEFINITIONS
 
+#if (__STDC_VERSION__ >= 199901L) /* C99 */ || \
+     (defined(__SUNPRO_C) && defined(__C99FEATURES__))
+#  define __FUNCTION__ __func__
+#else
+#  define __FUNCTION__ "<unknown>"
+#endif
+
 
 #if defined (WIN32) || defined (mingw32) || defined (cygwin)
 #define ENV_SEPARATOR ";"
@@ -1194,8 +1201,8 @@
 /* testing code sections. */
 #undef NETSNMP_ENABLE_TESTING_CODE
 
-/* Should evaluate to the name of the current function if defined */
-#undef NETSNMP_FUNCTION
+/* Should evaluate to the name of the current function if defined - disabled in Nexenta */
+#define NETSNMP_FUNCTION __FUNCTION__
 
 /* ifnet needs to have _KERNEL defined */
 #undef NETSNMP_IFNET_NEEDS_KERNEL
diff -urN net-snmp-5.5.upstream/perl/TrapReceiver/const-c.inc net-snmp-5.5/perl/TrapReceiver/const-c.inc
--- net-snmp-5.5.upstream/perl/TrapReceiver/const-c.inc	2009-01-01 02:26:43.000000000 +0530
+++ net-snmp-5.5/perl/TrapReceiver/const-c.inc	2011-02-22 10:06:10.000000000 +0530
@@ -87,11 +87,11 @@
 	       NETSNMPTRAPD_HANDLER_OK NETSNMPTRAPD_POST_HANDLER
 	       NETSNMPTRAPD_PRE_HANDLER));
 
-print constant_types(), "\n"; # macro defs
+print constant_types(); # macro defs
 foreach (C_constant ("NetSNMP::TrapReceiver", 'constant', 'IV', $types, undef, 3, @names) ) {
     print $_, "\n"; # C constant subs
 }
-print "\n#### XS Section:\n";
+print "#### XS Section:\n";
 print XS_constant ("NetSNMP::TrapReceiver", $types);
 __END__
    */
diff -urN net-snmp-5.5.upstream/perl/TrapReceiver/const-xs.inc net-snmp-5.5/perl/TrapReceiver/const-xs.inc
--- net-snmp-5.5.upstream/perl/TrapReceiver/const-xs.inc	2009-01-01 02:26:43.000000000 +0530
+++ net-snmp-5.5/perl/TrapReceiver/const-xs.inc	2011-02-22 10:06:10.000000000 +0530
@@ -22,14 +22,12 @@
            Second, if present, is found value */
         switch (type) {
         case PERL_constant_NOTFOUND:
-          sv =
-	    sv_2mortal(newSVpvf("%s is not a valid NetSNMP::TrapReceiver macro", s));
+          sv = sv_2mortal(newSVpvf("%s is not a valid NetSNMP::TrapReceiver macro", s));
           PUSHs(sv);
           break;
         case PERL_constant_NOTDEF:
           sv = sv_2mortal(newSVpvf(
-	    "Your vendor has not defined NetSNMP::TrapReceiver macro %s, used",
-				   s));
+	    "Your vendor has not defined NetSNMP::TrapReceiver macro %s, used", s));
           PUSHs(sv);
           break;
         case PERL_constant_ISIV:
