Fix the taboo list to avoid .dpkg-old, etc.



Index: logrotate-3.7.7/config.c
===================================================================
--- logrotate-3.7.7.orig/config.c	2008-11-07 02:13:37.003553893 +0000
+++ logrotate-3.7.7/config.c	2008-11-07 02:27:12.031554973 +0000
@@ -34,6 +34,7 @@
 #endif
 
 static char *defTabooExts[] = { ".rpmsave", ".rpmorig", "~", ",v",
+    ".disabled", ".dpkg-old", ".dpkg-dist", ".dpkg-new", ".cfsaved",
     ".rpmnew", ".swp", ".cfsaved", ".rhn-cfg-tmp-*"
 };
 static int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
