Index: logrotate-3.7.8/config.c
===================================================================
--- logrotate-3.7.8.orig/config.c	2009-08-14 23:21:35.024009420 +0100
+++ logrotate-3.7.8/config.c	2009-08-14 23:21:38.820009321 +0100
@@ -304,11 +304,6 @@
 	DIR *dirp;
 
 	here = open(".", O_RDONLY);
-	if (here < 0) {
-	    message(MESS_ERROR, "cannot open current directory: %s\n",
-		    strerror(errno));
-	    return 1;
-	}
 
 	if ((dirp = opendir(path)) == NULL) {
 	    message(MESS_ERROR, "cannot open directory %s: %s\n", path,
