Index: logrotate-3.7.7/logrotate.c
===================================================================
--- logrotate-3.7.7.orig/logrotate.c	2008-12-17 13:03:40.451635277 +0000
+++ logrotate-3.7.7/logrotate.c	2008-12-17 13:03:41.355635555 +0000
@@ -289,7 +289,7 @@
 		   char *uncompressCommand, char *address, char *subject)
 {
     int mailInput;
-    pid_t mailChild, uncompressChild;
+    pid_t mailChild, uncompressChild=-1;
     int mailStatus, uncompressStatus;
     int uncompressPipe[2];
     char *mailArgv[] = { mailCommand, "-s", subject, address, NULL };
