Index: b/main.c
===================================================================
--- a/main.c	2011-08-26 17:48:09.424798374 +0400
+++ b/main.c	2011-08-26 17:48:16.992497884 +0400
@@ -100,6 +100,7 @@
 # endif
 static struct sigaction sigact;
 #endif
+extern int default_af_hint;
 
 struct pid_status {
 	pid_t pid;
@@ -380,6 +381,23 @@
 			*t++ = '\0';
 		}
 
+#ifdef AF_INET
+                if (default_af_hint == AF_INET) {
+                    if (strncmp(cmd, "ssh", 3) == 0 || strstr(cmd, "/ssh") != NULL) {
+                        /* we're using ssh so we can add a -4 option */
+                        args[argc++] = "-4";
+                    }
+                }
+#endif
+#ifdef AF_INET6
+                if (default_af_hint == AF_INET6) {
+                    if (strncmp(cmd, "ssh", 3) == 0 || strstr(cmd, "/ssh") != NULL) {
+                        /* we're using ssh so we can add a -6 option */
+                        args[argc++] = "-6";
+                    }
+                }
+#endif
+
 		/* check to see if we've already been given '-l user' in
 		 * the remote-shell command */
 		for (i = 0; i < argc-1; i++) {
