Index: debhelper-8.1.6/Makefile
===================================================================
--- debhelper-8.1.6.orig/Makefile	2011-05-10 16:10:24.070139130 +0400
+++ debhelper-8.1.6/Makefile	2011-05-12 13:38:49.905752497 +0400
@@ -1,5 +1,5 @@
 # List of files of dh_* commands. Sorted for debhelper man page.
-COMMANDS=$(shell find . -maxdepth 1 -type f -perm +100 -name "dh_*" -printf "%f\n" | sort)
+COMMANDS=$(shell find . debian/nexenta -maxdepth 1 -type f -perm +100 -name "dh_*" | sort)
 
 # Find deprecated commands by looking at their synopsis.
 DEPRECATED=$(shell egrep -l '^dh_.* - .*deprecated' $(COMMANDS))
@@ -44,7 +44,7 @@
 LANGS=$(notdir $(basename $(wildcard man/po4a/po/*.po)))
 
 build: version debhelper.7
-	find . -maxdepth 1 -type f -perm +100 -name "dh*" \
+	find . debian/nexenta -maxdepth 1 -type f -perm +100 -name "dh*" \
 		-exec $(POD2MAN) {} {}.1 \;
 	po4a --previous -L UTF-8 man/po4a/po4a.cfg 
 	set -e; \
@@ -76,6 +76,7 @@
 	for lang in $(LANGS); do \
 		if [ -e man/$$lang ]; then rmdir man/$$lang; fi; \
 	done;
+	rm -f debian/nexenta/*.1
 
 install:
 	install -d $(DESTDIR)/usr/bin \
@@ -84,6 +85,7 @@
 		$(DESTDIR)$(PERLLIBDIR)/Buildsystem
 	install dh $(COMMANDS) $(DESTDIR)/usr/bin
 	install -m 0644 autoscripts/* $(DESTDIR)/usr/share/debhelper/autoscripts
+	install -m 0644 debian/nexenta/autoscripts/* $(DESTDIR)/usr/share/debhelper/autoscripts
 	install -m 0644 Debian/Debhelper/*.pm $(DESTDIR)$(PERLLIBDIR)
 	install -m 0644 Debian/Debhelper/Sequence/*.pm $(DESTDIR)$(PERLLIBDIR)/Sequence
 	install -m 0644 Debian/Debhelper/Buildsystem/*.pm $(DESTDIR)$(PERLLIBDIR)/Buildsystem
