Index: ntp_4.2.5p200/ntpd/ntp_io.c
===================================================================
--- ntp_4.2.5p200.orig/ntpd/ntp_io.c
+++ ntp_4.2.5p200/ntpd/ntp_io.c
@@ -1869,7 +1869,7 @@ enable_multicast_if(
 
 	case AF_INET:
 		if (setsockopt(iface->fd, IPPROTO_IP, IP_MULTICAST_IF,
-			       (char *)NSRCADR(&iface->sin),
+			       (void *)&NSRCADR(&iface->sin),
 			       sizeof(NSRCADR(&iface->sin)))) {
 
 			msyslog(LOG_ERR,
