# DP: Proper handling of packages in linecache.py

Index: b/Lib/linecache.py
===================================================================
--- a/Lib/linecache.py	2011-07-26 15:35:04.663280455 +0400
+++ b/Lib/linecache.py	2011-07-26 15:36:59.100154067 +0400
@@ -108,6 +108,11 @@
         if os.path.isabs(filename):
             return []
 
+        # Take care to handle packages
+        if basename == '__init__.py':
+            # filename referes to a package
+            basename = filename
+
         for dirname in sys.path:
             # When using imputil, sys.path may contain things other than
             # strings; ignore them when it happens.
