## disabledby-tetris.diff by Stefano Zacchiroli <zack@debian.org>

Index: vim-scripts/plugin/tetris.vim
===================================================================
--- vim-scripts.orig/plugin/tetris.vim
+++ vim-scripts/plugin/tetris.vim
@@ -8,6 +8,11 @@
 "	Dan Sharp		Bug reports, improvements
 "	If your name is not here, but should be, drop me a mail
 
+if exists("loaded_tetris")
+ finish
+endif
+let loaded_tetris = 1
+
 " TODO FocusGained FocusLost Auto calibration during play
 let s:s='-Tetris_game-'
 let s:WIDTH=10|let s:NEXTXPOS=16|let s:NEXTYPOS=2
