Index: vim-scripts/plugin/DoxygenToolkit.vim
===================================================================
--- vim-scripts.orig/plugin/DoxygenToolkit.vim
+++ vim-scripts/plugin/DoxygenToolkit.vim
@@ -226,10 +226,10 @@
 
 
 " Verify if already loaded
-"if exists("loaded_DoxygenToolkit")
-"	echo 'DoxygenToolkit Already Loaded.'
-"	finish
-"endif
+if exists("loaded_DoxygenToolkit")
+	echo 'DoxygenToolkit Already Loaded.'
+	finish
+endif
 let loaded_DoxygenToolkit = 1
 "echo 'Loading DoxygenToolkit...'
 let s:licenseTag = "Copyright (C) \<enter>\<enter>"
