diff -ur orig/dpkg-1.15.8.5stormos0/dselect/Makefile.in dpkg-1.15.8.5stormos0/dselect/Makefile.in
--- orig/dpkg-1.15.8.5stormos0/dselect/Makefile.in	2010-10-04 03:10:08.000000000 +0530
+++ dpkg-1.15.8.5stormos0/dselect/Makefile.in	2010-11-03 12:45:19.486764511 +0530
@@ -291,7 +291,7 @@
 	-DADMINDIR=\"$(admindir)\" -DLIBDIR=\"$(pkglibdir)\" \
 	-DLOCALLIBDIR=\"/usr/local/lib/dpkg\" \
 	-idirafter $(top_srcdir)/lib/compat \
-	-iquote $(builddir) \
+	-I$(builddir) \
 	-I$(top_builddir) \
 	-I$(top_srcdir)/lib
 
