diff -ur dpkg-1.15.8.5.old/lib/dpkg/database.c dpkg-1.15.8.5/lib/dpkg/database.c
--- dpkg-1.15.8.5.old/lib/dpkg/database.c	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/database.c	2011-03-21 17:08:00.000000000 -0700
@@ -72,6 +72,7 @@
   blankversion(&pigp->configversion);
   pigp->files= NULL;
   pigp->clientdata= NULL;
+  pigp->zone= NULL;
   pigp->trigaw.head = pigp->trigaw.tail = NULL;
   pigp->othertrigaw_head = NULL;
   pigp->trigpend_head = NULL;
diff -ur dpkg-1.15.8.5.old/lib/dpkg/dpkg-db.h dpkg-1.15.8.5/lib/dpkg/dpkg-db.h
--- dpkg-1.15.8.5.old/lib/dpkg/dpkg-db.h	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/dpkg-db.h	2011-03-21 17:08:00.000000000 -0700
@@ -170,6 +170,7 @@
   } priority;
   const char *otherpriority;
   const char *section;
+  const char *zone;
   struct versionrevision configversion;
   struct filedetails *files;
   struct pkginfoperfile installed;
diff -ur dpkg-1.15.8.5.old/lib/dpkg/dump.c dpkg-1.15.8.5/lib/dpkg/dump.c
--- dpkg-1.15.8.5.old/lib/dpkg/dump.c	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/dump.c	2011-03-21 17:08:00.000000000 -0700
@@ -98,6 +98,18 @@
     varbufaddc(vb,'\n');
 }
 
+void w_zone(struct varbuf *vb,
+               const struct pkginfo *pigp, const struct pkginfoperfile *pifp,
+               enum fwriteflags flags, const struct fieldinfo *fip) {
+  const char *value= pigp->zone;
+  if (!value || !*value) return;
+  if (flags&fw_printheader)
+    varbufaddstr(vb,"Zone: ");
+  varbufaddstr(vb,value);
+  if (flags&fw_printheader)
+    varbufaddc(vb,'\n');
+}
+
 void w_charfield(struct varbuf *vb,
                  const struct pkginfo *pigp, const struct pkginfoperfile *pifp,
                  enum fwriteflags flags, const struct fieldinfo *fip) {
diff -ur dpkg-1.15.8.5.old/lib/dpkg/fields.c dpkg-1.15.8.5/lib/dpkg/fields.c
--- dpkg-1.15.8.5.old/lib/dpkg/fields.c	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/fields.c	2011-03-21 17:08:00.000000000 -0700
@@ -212,6 +212,13 @@
 
 }
 
+void f_zone(struct pkginfo *pigp, struct pkginfoperfile *pifp,
+                struct parsedb_state *ps,
+                const char *value, const struct fieldinfo *fip) {
+  if (!*value) return;
+  pigp->zone= nfstrsave(value);
+}
+
 static void conffvalue_lastword(const char *value, const char *from,
                                 const char *endent,
                                 const char **word_start_r, int *word_len_r,
diff -ur dpkg-1.15.8.5.old/lib/dpkg/parse.c dpkg-1.15.8.5/lib/dpkg/parse.c
--- dpkg-1.15.8.5.old/lib/dpkg/parse.c	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/parse.c	2011-03-10 23:55:50.000000000 -0800
@@ -76,6 +76,7 @@
   { "Description",      f_charfield,       w_charfield,      PKGIFPOFF(description)   },
   { "Triggers-Pending", f_trigpend,        w_trigpend                                 },
   { "Triggers-Awaited", f_trigaw,          w_trigaw                                   },
+  { "Zone",           	f_zone,            w_zone                                   },
   /* Note that aliases are added to the nicknames table in parsehelp.c. */
   {  NULL   /* sentinel - tells code that list is ended */                               }
 };
@@ -348,6 +349,11 @@
         !((flags & pdb_weakclassification) && pigp->priority != pri_unknown)) {
       pigp->priority= newpig.priority;
       if (newpig.priority == pri_other) pigp->otherpriority= newpig.otherpriority;
+
+    if (newpig.zone && *newpig.zone &&
+        !((flags & pdb_weakclassification) && pigp->zone && *pigp->zone))
+      pigp->zone= newpig.zone;
+
     }
 
     /* Sort out the dependency mess. */
diff -ur dpkg-1.15.8.5.old/lib/dpkg/parsedump.h dpkg-1.15.8.5/lib/dpkg/parsedump.h
--- dpkg-1.15.8.5.old/lib/dpkg/parsedump.h	2010-11-03 22:35:47.000000000 -0700
+++ dpkg-1.15.8.5/lib/dpkg/parsedump.h	2011-03-21 17:08:00.000000000 -0700
@@ -44,6 +44,7 @@
 freadfunction f_name, f_charfield, f_priority, f_section, f_status, f_filecharf;
 freadfunction f_boolean, f_dependency, f_conffiles, f_version, f_revision;
 freadfunction f_configversion;
+freadfunction f_zone;
 freadfunction f_trigpend, f_trigaw;
 
 enum fwriteflags {
@@ -56,6 +57,7 @@
 fwritefunction w_name, w_charfield, w_priority, w_section, w_status, w_configversion;
 fwritefunction w_version, w_null, w_booleandefno, w_dependency, w_conffiles;
 fwritefunction w_filecharf;
+fwritefunction w_zone;
 fwritefunction w_trigpend, w_trigaw;
 
 struct fieldinfo {
diff -ur dpkg-1.15.8.5.old/src/filesdb.c dpkg-1.15.8.5/src/filesdb.c
--- dpkg-1.15.8.5.old/src/filesdb.c	2010-11-03 22:36:07.000000000 -0700
+++ dpkg-1.15.8.5/src/filesdb.c	2011-03-23 13:55:32.000000000 -0700
@@ -272,7 +272,8 @@
     onerr_abort--;
     if (pkg->status != stat_configfiles) {
       if (saidread == 1) putc('\n',stderr);
-      warning(_("files list file for package `%.250s' missing, assuming "
+      if (!((pkg->zone !=NULL) && strncmp(pkg->zone,"global", strlen("global"))==0))
+           warning(_("files list file for package `%.250s' missing, assuming "
                 "package has no files currently installed."), pkg->name);
     }
     pkg->clientdata->files = NULL;
diff -ur dpkg-1.15.8.5.old/src/main.c dpkg-1.15.8.5/src/main.c
--- dpkg-1.15.8.5.old/src/main.c	2010-11-03 22:36:07.000000000 -0700
+++ dpkg-1.15.8.5/src/main.c	2011-03-21 17:08:00.000000000 -0700
@@ -190,6 +190,7 @@
 int errabort = 50;
 const char *admindir= ADMINDIR;
 const char *instdir= "";
+const char *current_zone = NULL;
 struct pkg_list *ignoredependss = NULL;
 
 static const struct forceinfo {
diff -ur dpkg-1.15.8.5.old/src/main.h dpkg-1.15.8.5/src/main.h
--- dpkg-1.15.8.5.old/src/main.h	2010-11-03 22:36:07.000000000 -0700
+++ dpkg-1.15.8.5/src/main.h	2011-03-21 22:25:04.000000000 -0700
@@ -137,8 +137,10 @@
 extern int errabort;
 extern const char *admindir;
 extern const char *instdir;
+extern const char *current_zone;
 extern struct pkg_list *ignoredependss;
 extern const char architecture[];
+extern bool hollow_install(char *admindir, char *pkg_zone);
 
 struct invoke_hook {
 	struct invoke_hook *next;
diff -ur dpkg-1.15.8.5.old/src/processarc.c dpkg-1.15.8.5/src/processarc.c
--- dpkg-1.15.8.5.old/src/processarc.c	2010-11-03 22:36:07.000000000 -0700
+++ dpkg-1.15.8.5/src/processarc.c	2011-03-21 22:25:14.000000000 -0700
@@ -124,6 +124,8 @@
   struct pkg_deconf_list *deconpil, *deconpiltemp;
   struct rename_list *rename_head = NULL, *rename_node = NULL;
   
+  bool hollow = false;
+  
   cleanup_pkg_failed= cleanup_conflictor_failed= 0;
   admindirlen= strlen(admindir);
 
@@ -239,6 +241,11 @@
 
   parsedb(cidir, pdb_recordavailable | pdb_rejectstatus | pdb_ignorefiles,
           &pkg,NULL,NULL);
+  
+  /* check zone */
+  if (hollow_install(admindir, pkg->zone))
+    hollow = true;
+  
   if (!pkg->files) {
     pkg->files= nfmalloc(sizeof(struct filedetails));
     pkg->files->next = NULL;
@@ -613,6 +620,8 @@
    * files get replaced `as we go'.
    */
 
+  if (!hollow) { /* in non-global zone we shouldn't unpack archive */
+  
   m_pipe(p1);
   push_cleanup(cu_closepipe, ehflag_bombout, NULL, 0, 1, (void *)&p1[0]);
   c1 = subproc_fork();
@@ -931,6 +940,8 @@
 
   pop_cleanup(ehflag_normaltidy); /* closedir */
 
+  } /* end of hollow install */ 
+  
   /* Update the status database.
    * This involves copying each field across from the `available'
    * to the `installed' half of the pkg structure.
diff -ur /dev/null dpkg-1.15.8.5/src/zone.c
--- /dev/null			2011-06-11 14:50:55.000000000 +0100
+++ dpkg-1.15.8.5/src/zone.c	2011-03-22 01:01:10.000000000 +0000
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2011 Nexenta Systems, Inc.  All rights reserved.
+ */
+	  
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <instzones_api.h>
+
+bool
+hollow_install(char *admindir, char *pkg_zone) 
+{
+    int k;
+    char *current_zone = NULL;
+    char *zonename = NULL;
+    char *zonepath = NULL;
+    bool hollow = false;
+
+    zoneList_t zlst;
+    zoneBrandList_t *brands = NULL;
+
+    current_zone = z_get_zonename();
+
+    if ((current_zone!=NULL) && (*current_zone !='\0'))
+	    printf("Processing in zone: %s\n",current_zone);
+    
+    if ((pkg_zone != NULL) && strncmp(pkg_zone,"global",strlen("global")) == 0) { 
+	    if (z_running_in_global_zone()) {
+		    if ((brands = z_make_brand_list("native cluster nexenta"," ")) == NULL) {
+			    return false;
+		    }
+		    zlst = z_get_nonglobal_zone_list_by_brand(brands);
+		    if (zlst == (zoneList_t)NULL) {
+			    hollow = false;
+		    } else {
+			    for (k=0; (zonename=z_zlist_get_zonename(zlst,k)) != NULL; k++) {
+				if (z_zlist_get_current_state(zlst,k) == 
+				    ZONE_STATE_INSTALLED || z_zlist_get_current_state(zlst,k) == 
+				    ZONE_STATE_RUNNING || z_zlist_get_current_state(zlst,k) == 
+				    ZONE_STATE_DOWN) {
+					zonepath = z_zlist_get_zonepath(zlst,k);
+					/* admindir in zone, hollow installation */
+					if (strstr(admindir, zonepath) != NULL)
+						hollow = true;
+		    		}
+			    }
+		    }    
+		    z_free_brand_list(brands);
+		    z_free_zone_list(zlst);
+	    } else {
+		    /* non-global zone, hollow installation */
+		    hollow = true;
+	    }	
+    }
+
+    return hollow;
+}
diff -ur dpkg-1.15.8.5.old/src/Makefile.am dpkg-1.15.8.5/src/Makefile.am
--- dpkg-1.15.8.5.old/src/Makefile.am	2011-01-30 19:37:44.000000000 +0000
+++ dpkg-1.15.8.5/src/Makefile.am	2011-03-21 23:46:58.000000000 +0000
@@ -39,12 +39,14 @@
 	remove.c \
 	select.c \
 	trigproc.c \
-	update.c
+	update.c \
+	zone.c
 
 dpkg_LDADD = \
 	../lib/dpkg/libdpkg.a \
 	../lib/compat/libcompat.a \
 	$(LIBINTL) \
+	$(INSTZONES_LIBS) \
 	$(SELINUX_LIBS)
 
 dpkg_divert_SOURCES = \
diff -ur dpkg-1.15.8.5.old/configure.ac dpkg-1.15.8.5/src/configure.ac
--- dpkg-1.15.8.5.old/configure.ac	2011-01-30 19:37:43.000000000 +0000
+++ dpkg-1.15.8.5/configure.ac		2011-06-11 15:15:10.522428413 +0100
@@ -100,6 +100,7 @@
 DPKG_LIB_ZLIB
 DPKG_LIB_BZ2
 DPKG_LIB_SELINUX
+DPKG_LIB_INSTZONES
 if test "x$build_dselect" = "xyes"; then
    DPKG_LIB_CURSES
 fi
diff -ur dpkg-1.15.8.5.old/m4/dpkg-libs.m4 dpkg-1.15.8.5/m4/dpkg-libs.m4
--- dpkg-1.15.8.5.old/m4/dpkg-libs.m4	2011-06-11 15:28:53.175992260 +0100
+++ dpkg-1.15.8.5/m4/dpkg-libs.m4	2011-06-11 15:35:22.632788802 +0100
@@ -49,6 +49,16 @@
   DPKG_WITH_COMPRESS_LIB([bz2], [bzlib.h], [BZ2_bzdopen], [bz2])
 ])# DPKG_LIB_BZ2
 
+# DPKG_LIB_INSTZONES
+# ------------------
+# Check for instzones library.
+AC_DEFUN([DPKG_LIB_INSTZONES],
+  [AC_ARG_VAR([INSTZONES_LIBS], [linker flags for zones support])
+  AC_CHECK_LIB([instzones], [z_get_zonename],
+  [INSTZONES_LIBS="${INSTZONES_LIBS:+$INSTZONES_LIBS }-linstzones"],
+  [AC_MSG_FAILURE([no instzones library found])])
+])# DPKG_LIB_INSTZONES
+
 # DPKG_LIB_SELINUX
 # ----------------
 # Check for selinux library.
