Index: debhelper-8.9.13~bpo60+1/Debian/Debhelper/Dh_Lib.pm
===================================================================
--- a/Debian/Debhelper/Dh_Lib.pm
+++ b/Debian/Debhelper/Dh_Lib.pm
@@ -96,6 +96,12 @@ sub init {
 	if (! exists $dh{ERROR_HANDLER} || ! defined $dh{ERROR_HANDLER}) {
 		$dh{ERROR_HANDLER}='exit \$?';
 	}
+
+    if ($ENV{'PATH'} =~ /^\./) {
+        $ENV{'PATH'} = '.:/usr/gnu/bin:' . $ENV{'PATH'};
+    } else {
+        $ENV{'PATH'} = '/usr/gnu/bin:' . $ENV{'PATH'};
+    }
 }
 
 # Run at exit. Add the command to the log files for the packages it acted
