Index: b/support/rsyncstats
===================================================================
--- a/support/rsyncstats	2011-08-26 17:48:09.802427662 +0400
+++ b/support/rsyncstats	2011-08-26 17:48:15.756252127 +0400
@@ -12,7 +12,7 @@
 use Getopt::Long;
 
 # You may wish to edit the next line to customize for your default log file.
-$usage_file = "/var/log/rsyncd.log";
+$usage_file = "/var/log/rsyncd";
 
 # Edit the following lines for default report settings.
 # Entries defined here will be over-ridden by the command line.
