Index: coreutils-8.5/configure
===================================================================
--- coreutils-8.5.orig/configure
+++ coreutils-8.5/configure
@@ -53276,24 +53276,20 @@ rm -f conftest*
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $elf_sys" >&5
 $as_echo "$elf_sys" >&6; }
-if test "$elf_sys" = "yes" && \
-   test "$GCC" = "yes"; then
-  {
-  if test -z "$optional_bin_progs"; then
-    optional_bin_progs=stdbuf
-  else
-    optional_bin_progs="$optional_bin_progs stdbuf"
-  fi
-}
-  {
-  if test -z "$optional_pkglib_progs"; then
-    optional_pkglib_progs=libstdbuf.so
-  else
-    optional_pkglib_progs="$optional_pkglib_progs libstdbuf.so"
-  fi
-}
+
+if test -z "$optional_bin_progs"; then
+  optional_bin_progs=stdbuf
+else
+  optional_bin_progs="$optional_bin_progs stdbuf"
 fi
 
+if test -z "$optional_pkglib_progs"; then
+  optional_pkglib_progs=libstdbuf.so
+else
+  optional_pkglib_progs="$optional_pkglib_progs libstdbuf.so"
+fi
+
+
 ############################################################################
 mk="$srcdir/src/Makefile.am"
 # Extract all literal names from the definition of $(EXTRA_PROGRAMS)
