Fix the taboo list to avoid UCF files.



Index: logrotate-3.7.8/config.c
===================================================================
--- logrotate-3.7.8.orig/config.c	2008-11-07 02:13:37.003553893 +0000
+++ logrotate-3.7.8/config.c	2010-04-17 16:27:12.031554973 +4000
@@ -35,6 +35,7 @@
 
 static char *defTabooExts[] = { ".rpmsave", ".rpmorig", "~", ",v",
     ".disabled", ".dpkg-old", ".dpkg-dist", ".dpkg-new", ".cfsaved",
+    ".ucf-old", ".ucf-dist", ".ucf-new",
     ".rpmnew", ".swp", ".cfsaved", ".rhn-cfg-tmp-*"
 };
 static int defTabooCount = sizeof(defTabooExts) / sizeof(char *);
