--- watch-3.2.7.orig/watch.c	2006-06-17 02:18:38.000000000 -0700
+++ watch-3.2.7/watch.c	2013-07-15 21:23:21.635845625 -0700
@@ -24,7 +24,6 @@
 #include <unistd.h>
 #include <termios.h>
 #include <locale.h>
-#include "proc/procps.h"
 
 #ifdef FORCE_8BIT
 #undef isprint
@@ -53,14 +52,14 @@
 
 #define min(x,y) ((x) > (y) ? (y) : (x))
 
-static void do_usage(void) NORETURN;
+static void do_usage(void);
 static void do_usage(void)
 {
 	fprintf(stderr, usage, progname);
 	exit(1);
 }
 
-static void do_exit(int status) NORETURN;
+static void do_exit(int status);
 static void do_exit(int status)
 {
 	if (curses_started)
@@ -69,7 +68,7 @@
 }
 
 /* signal handler */
-static void die(int notused) NORETURN;
+static void die(int notused);
 static void die(int notused)
 {
 	(void) notused;
