diff -urN Python-2.6.9.old/configure Python-2.6.9/configure
--- Python-2.6.9.old/configure	2013-10-29 08:04:39.000000000 -0700
+++ Python-2.6.9/configure	2014-10-15 05:20:07.021245498 -0700
@@ -11422,7 +11422,7 @@
 fi
 
 cat >>confdefs.h <<_ACEOF
-#define HAVE_DECL_ISFINITE $ac_have_decl
+#define HAVE_DECL_ISFINITE 0
 _ACEOF
 
 
diff -urN Python-2.6.9.old/configure.in Python-2.6.9/configure.in
--- Python-2.6.9.old/configure.in	2013-10-29 08:04:39.000000000 -0700
+++ Python-2.6.9/configure.in	2014-10-15 05:23:08.553374818 -0700
@@ -3370,7 +3370,7 @@
 fi
 
 AC_CHECK_FUNCS([acosh asinh atanh copysign expm1 finite hypot log1p])
-AC_CHECK_DECLS([isinf, isnan, isfinite], [], [], [[#include <math.h>]])
+AC_CHECK_DECLS([isinf, isnan], [], [])
 
 LIBS=$LIBS_SAVE
 
