Index: libapache2-mod-encoding-20040616/mod_encoding.c
===================================================================
--- libapache2-mod-encoding-20040616.orig/mod_encoding.c	2011-05-24 01:43:35.404896652 +0400
+++ libapache2-mod-encoding-20040616/mod_encoding.c	2011-05-24 01:43:39.784682249 +0400
@@ -353,23 +353,23 @@
 static const command_rec mod_enc_commands[] = {
   {"EncodingEngine",
    set_encoding_engine, NULL,
-   OR_ALL, FLAG,  "Usage: EncodingEngine (on|off)"},
+   RSRC_CONF, FLAG,  "Usage: EncodingEngine (on|off)"},
 
   {"SetServerEncoding",
    set_server_encoding, NULL,
-   OR_ALL, TAKE1, "Usage: SetServerEncoding <enc>"},
+   RSRC_CONF, TAKE1, "Usage: SetServerEncoding <enc>"},
 
   {"AddClientEncoding",
    add_client_encoding, NULL,
-   OR_ALL, RAW_ARGS, "Usage: AddClientEncoding <agent> <enc> [<enc> ...]"},
+   RSRC_CONF, RAW_ARGS, "Usage: AddClientEncoding <agent> <enc> [<enc> ...]"},
 
   {"DefaultClientEncoding",
    default_client_encoding, NULL,
-   OR_ALL, RAW_ARGS, "Usage: DefaultClientEncoding <enclist>"},
+   RSRC_CONF, RAW_ARGS, "Usage: DefaultClientEncoding <enclist>"},
 
   {"NormalizeUsername",
    set_normalize_username, NULL,
-   OR_ALL, FLAG, "Usage: NormalizeUsername (on|off)"},
+   RSRC_CONF|ACCESS_CONF, FLAG, "Usage: NormalizeUsername (on|off)"},
 
   {NULL}
 };
