Index: libio-capture-perl/lib/IO/Capture/Overview.pod
===================================================================
--- libio-capture-perl.orig/lib/IO/Capture/Overview.pod	2008-01-28 00:11:36.000000000 -0600
+++ libio-capture-perl/lib/IO/Capture/Overview.pod	2008-01-28 00:13:41.000000000 -0600
@@ -1,6 +1,6 @@
 =head1 NAME
 
-Overview of C<IO::Capture> Module, and classes derived from it. 
+IO::Capture -- Overview of C<IO::Capture> Module, and classes derived from it.
 
 =head1 DESCRIPTION
 
Index: libio-capture-perl/lib/IO/Capture/Tie_STDx.pm
===================================================================
--- libio-capture-perl.orig/lib/IO/Capture/Tie_STDx.pm	2008-01-28 00:12:18.000000000 -0600
+++ libio-capture-perl/lib/IO/Capture/Tie_STDx.pm	2008-01-28 00:13:09.000000000 -0600
@@ -28,7 +28,7 @@
 
 =head1 NAME
 
-IO::Capture::Tie_STDx;
+IO::Capture::Tie_STDx -- tie STDOUT or STDERR;
 
 =head1 SYNOPSIS
 
