Index: redis-2.2.7/src/Makefile
===================================================================
--- redis-2.2.7.orig/src/Makefile	2011-05-20 19:15:31.037913928 +0400
+++ redis-2.2.7/src/Makefile	2011-05-20 19:16:58.153230295 +0400
@@ -2,17 +2,6 @@
 # Copyright (C) 2009 Salvatore Sanfilippo <antirez at gmail dot com>
 # This file is released under the BSD license, see the COPYING file
 
-uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
-OPTIMIZATION?=-O2
-ifeq ($(uname_S),SunOS)
-  CFLAGS?= -std=c99 -pedantic $(OPTIMIZATION) -Wall -W -D__EXTENSIONS__ -D_XPG6
-  CCLINK?= -ldl -lnsl -lsocket -lm -lpthread
-  DEBUG?= -g -ggdb 
-else
-  CFLAGS?= -std=c99 -pedantic $(OPTIMIZATION) -Wall -W $(ARCH) $(PROF)
-  CCLINK?= -lm -pthread
-  DEBUG?= -g -rdynamic -ggdb 
-endif
 
 ifeq ($(USE_TCMALLOC),yes)
   CCLINK+= -ltcmalloc
@@ -110,26 +99,26 @@
 	cd ../deps/linenoise && $(MAKE) ARCH="$(ARCH)"
 
 redis-server: $(OBJ)
-	$(CC) -o $(PRGNAME) $(CCOPT) $(DEBUG) $(OBJ)
+	$(CC) -o $(PRGNAME) $(CFLAGS) $(LDFLAGS) $(OBJ)
 
 redis-benchmark: dependencies $(BENCHOBJ)
 	cd ../deps/hiredis && $(MAKE) static
-	$(CC) -o $(BENCHPRGNAME) $(CCOPT) $(DEBUG) $(BENCHOBJ) ../deps/hiredis/libhiredis.a
+	$(CC) -o $(BENCHPRGNAME) $(CFLAGS) $(LDFLAGS) $(BENCHOBJ) ../deps/hiredis/libhiredis.a
 
 redis-benchmark.o:
 	$(CC) -c $(CFLAGS) -I../deps/hiredis $(DEBUG) $(COMPILE_TIME) $<
 
 redis-cli: dependencies $(CLIOBJ)
-	$(CC) -o $(CLIPRGNAME) $(CCOPT) $(DEBUG) $(CLIOBJ) ../deps/hiredis/libhiredis.a ../deps/linenoise/linenoise.o
+	$(CC) -o $(CLIPRGNAME) $(CFLAGS) $(LDFLAGS) $(CLIOBJ) ../deps/hiredis/libhiredis.a ../deps/linenoise/linenoise.o
 
 redis-cli.o:
 	$(CC) -c $(CFLAGS) -I../deps/hiredis -I../deps/linenoise $(DEBUG) $(COMPILE_TIME) $<
 
 redis-check-dump: $(CHECKDUMPOBJ)
-	$(CC) -o $(CHECKDUMPPRGNAME) $(CCOPT) $(DEBUG) $(CHECKDUMPOBJ)
+	$(CC) -o $(CHECKDUMPPRGNAME) $(CFLAGS) $(LDFLAGS) $(CHECKDUMPOBJ)
 
 redis-check-aof: $(CHECKAOFOBJ)
-	$(CC) -o $(CHECKAOFPRGNAME) $(CCOPT) $(DEBUG) $(CHECKAOFOBJ)
+	$(CC) -o $(CHECKAOFPRGNAME) $(CFLAGS) $(LDFLAGS) $(CHECKAOFOBJ)
 
 .c.o:
 	$(CC) -c $(CFLAGS) $(DEBUG) $(COMPILE_TIME) $<
Index: redis-2.2.7/deps/hiredis/Makefile
===================================================================
--- redis-2.2.7.orig/deps/hiredis/Makefile	2011-05-20 19:15:30.775714342 +0400
+++ redis-2.2.7/deps/hiredis/Makefile	2011-05-20 19:16:58.160370873 +0400
@@ -8,9 +8,9 @@
 uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
 OPTIMIZATION?=-O3
 ifeq ($(uname_S),SunOS)
-  CFLAGS?=-std=c99 -pedantic $(OPTIMIZATION) -fPIC -Wall -W -D__EXTENSIONS__ -D_XPG6 $(ARCH) $(PROF)
-  CCLINK?=-ldl -lnsl -lsocket -lm -lpthread
-  LDFLAGS?=-L. -Wl,-R,.
+  CFLAGS+=-std=c99 -pedantic $(OPTIMIZATION) -fPIC -Wall -W -D__EXTENSIONS__ -D_XPG6 $(ARCH) $(PROF)
+  CCLINK+=-ldl -lnsl -lsocket -lm -lpthread
+  LDFLAGS+=-L. -Wl,-R,.
   DYLIBNAME?=libhiredis.so
   DYLIB_MAKE_CMD?=$(CC) -G -o ${DYLIBNAME} ${OBJ}
   STLIBNAME?=libhiredis.a
Index: redis-2.2.7/deps/linenoise/Makefile
===================================================================
--- redis-2.2.7.orig/deps/linenoise/Makefile	2011-05-20 19:17:58.719727439 +0400
+++ redis-2.2.7/deps/linenoise/Makefile	2011-05-20 19:18:57.596740163 +0400
@@ -1,10 +1,10 @@
 linenoise_example: linenoise.h linenoise.c
 
 linenoise_example: linenoise.o example.o
-	$(CC) $(ARCH) -Wall -W -Os -g -o linenoise_example linenoise.o example.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o linenoise_example linenoise.o example.o
 
 .c.o:
-	$(CC) $(ARCH) -c -Wall -W -Os -g $<
+	$(CC) $(CFLAGS) -c $<
 
 clean:
 	rm -f linenoise_example *.o
