Index: b/bus/config-parser.c
===================================================================
--- a/bus/config-parser.c	2010-07-14 00:43:38.000000000 +0200
+++ b/bus/config-parser.c	2010-07-17 15:16:44.226404084 +0200
@@ -2236,8 +2236,18 @@
         {
           if (!include_file (parser, &full_path, TRUE, error))
             {
-              _dbus_string_free (&full_path);
-              goto failed;
+              /* Debian patch to skip malformed /etc/dbus-1/system.d entries */
+              /*
+               * _dbus_string_free (&full_path);
+               * goto failed;
+               */
+              if (dbus_error_is_set (error))
+                {
+                  _dbus_warn("\nEncountered error '%s' while parsing '%s'\n",
+                             error->message,
+                             _dbus_string_get_const_data(&full_path));
+                  dbus_error_free (error);
+                }
             }
         }
 
