Index: dbus-1.4.0-2/configure.in
===================================================================
--- dbus-1.4.0-2.orig/configure.in	2011-04-28 14:44:49.742912348 +0400
+++ dbus-1.4.0-2/configure.in	2011-04-28 14:45:33.112880001 +0400
@@ -1171,19 +1171,6 @@
      ;;
   esac
 
-  case " $CFLAGS " in
-  *[\ \	]-fPIE[\ \	]*) ;;
-  *) if test x$dbus_win = xno && cc_supports_flag -fPIE; then
-        PIE_CFLAGS="-fPIE"
-        if ld_supports_flag -z,relro; then
-           PIE_LDFLAGS="-pie -Wl,-z,relro"
-        else
-           PIE_LDFLAGS="-pie"
-        fi
-     fi
-     ;;
-  esac
-
   ### Disabled warnings, and compiler flag overrides
 
   # Let's just ignore unused for now
