Description: Use mailcap entry to choose the images viewer
Bug-Debian: http://bugs.debian.org/544058
Author: Andreas Tille <tillea@rki.de>

--- mc-4.7.0-pre4.orig/misc/mc.ext.in	2009-12-17 19:51:51.000000000 +0100
+++ mc-4.7.0-pre4/misc/mc.ext.in	2009-12-17 19:53:00.000000000 +0100
@@ -387,7 +387,7 @@
 	Include=image
 
 include/image
-	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	Open=(see %f &)
 	View=%view{ascii} identify %f
 	#View=%view{ascii} asciiview %f
 
