--- sshpass-1.05/main.c.orig	Mon Jun  2 10:01:25 2014
+++ sshpass-1.05/main.c	Mon Jun  2 10:26:42 2014
@@ -358,9 +358,10 @@
 {
     // We are looking for the string
     static int prevmatch=0; // If the "password" prompt is repeated, we have the wrong password.
-    static int state1, state2;
+    static int state1, state2, state3;
     static const char compare1[]="assword:"; // Asking for a password
     static const char compare2[]="The authenticity of host "; // Asks to authenticate host
+    static const char compare3[]= "assword for "; // Asking for the OpenPAM password
     // static const char compare3[]="WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!"; // Warns about man in the middle attack
     // The remote identification changed error is sent to stderr, not the tty, so we do not handle it.
     // This is not a problem, as ssh exists immediately in such a case
@@ -383,6 +384,21 @@
 	}
     }
 
+    // Are we at a OpenPAM password prompt?
+    if( ret==0 ) {
+        state3=match( compare3, buffer, numread, state3 );
+        if( compare3[state3]=='\0' ) {
+            if( !prevmatch ) {
+                write_pass( fd );
+                state3=0;
+                prevmatch=1;
+            } else {
+                // Wrong password - terminate with proper error code
+                ret=RETURN_INCORRECT_PASSWORD;
+            }
+        }
+    }
+
     if( ret==0 ) {
         state2=match( compare2, buffer, numread, state2 );
 
