From: Ansgar Burchardt <ansgar@43-1.org>
Date: Tue, 23 Mar 2010 20:47:55 +0900
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=55836
Subject: Fix spelling error

--- libhtml-tree-perl.orig/lib/HTML/AsSubs.pm
+++ libhtml-tree-perl/lib/HTML/AsSubs.pm
@@ -22,7 +22,7 @@
 
 This module exports functions that can be used to construct various
 HTML elements. The functions are named after the tags of the
-correponding HTML element and are all written in lower case. If the
+corresponding HTML element and are all written in lower case. If the
 first argument is a hash reference then it will be used to initialize the
 attributes of this element. The remaining arguments are regarded as
 content.
--- libhtml-tree-perl.orig/lib/HTML/Tree/AboutObjects.pod
+++ libhtml-tree-perl/lib/HTML/Tree/AboutObjects.pod
@@ -562,7 +562,7 @@
 into limitations with others.  For example, a Business::US_Amort
 object can be turned into a string with Data::Dumper, and that string
 written to a file.  When it's restored later, its attributes will be
-accessable as normal.  But in the unlikely case that the loan object was
+accessible as normal.  But in the unlikely case that the loan object was
 saved in mid-calculation, the calculation may not be resumable.  This is
 because of the way that that I<particular> class does its calculations,
 but similar limitations may occur with objects from other classses.
--- libhtml-tree-perl.orig/lib/HTML/Tree/AboutTrees.pod
+++ libhtml-tree-perl/lib/HTML/Tree/AboutTrees.pod
@@ -670,8 +670,8 @@
 For the computer to decide which of these is the best move to make, it
 needs to quantify the benefit of these moves as a number -- call that
 the "payoff".  The payoff of a move can be figured as just the number
-of x pieces removed by the most recent move, minus the nubmer of o
-pieces removed by the mots recent move.  (It so happens that the rules
+of x pieces removed by the most recent move, minus the number of o
+pieces removed by the most recent move.  (It so happens that the rules
 of the game mean that no move can delete both o's and x's, but the
 formula still applies.)  Since none of these moves removed any pieces,
 all these moves have the same immediate payoff: 0.
--- libhtml-tree-perl.orig/lib/HTML/Element.pm
+++ libhtml-tree-perl/lib/HTML/Element.pm
@@ -3042,7 +3042,7 @@
   for (@{  $e->extract_links('a', 'img')  }) {
       my($link, $element, $attr, $tag) = @$_;
       print
-        "Hey, there's a $tag that links to "
+        "Hey, there's a $tag that links to ",
         $link, ", in its $attr attribute, at ",
         $element->address(), ".\n";
   }
