diff -urN DTraceToolkit-0.99/Bin/dtruss DTraceToolkit-0.99.new/Bin/dtruss
--- DTraceToolkit-0.99/Bin/dtruss	2007-08-07 14:21:07.000000000 +0400
+++ DTraceToolkit-0.99.new/Bin/dtruss	2012-11-29 16:45:05.530696539 +0400
@@ -387,7 +387,7 @@
 }
 
 /* print 2 arg output */
-syscall::utime:return,
+syscall::utimesys:return,
 syscall::munmap:return
 /self->start/
 {
diff -urN DTraceToolkit-0.99/dtruss DTraceToolkit-0.99.new/dtruss
--- DTraceToolkit-0.99/dtruss	2007-08-07 14:21:07.000000000 +0400
+++ DTraceToolkit-0.99.new/dtruss	2012-11-29 16:45:05.530696539 +0400
@@ -387,7 +387,7 @@
 }
 
 /* print 2 arg output */
-syscall::utime:return,
+syscall::utimesys:return,
 syscall::munmap:return
 /self->start/
 {
