Index: openldap-2.4.21/servers/slapd/overlays/Makefile.in
===================================================================
--- openldap-2.4.21.orig/servers/slapd/overlays/Makefile.in	2011-04-05 16:40:16.738972865 +0400
+++ openldap-2.4.21/servers/slapd/overlays/Makefile.in	2011-04-05 16:40:33.043605111 +0400
@@ -102,7 +102,7 @@
 	$(LTLINK_MOD) -module -o $@ retcode.lo version.lo $(LINK_LIBS)
 
 rwm_x.o: rwm.o rwmconf.o rwmdn.o rwmmap.o
-	$(LD) -r -o $@ rwm.o rwmconf.o rwmdn.o rwmmap.o
+	$(LD) $(LDFLAGS64) -r -o $@ rwm.o rwmconf.o rwmdn.o rwmmap.o
 
 rwm.la : rwm.lo rwmconf.lo rwmdn.lo rwmmap.lo
 	$(LTLINK_MOD) -module -o $@ rwm.lo rwmconf.lo rwmdn.lo rwmmap.lo version.lo $(LINK_LIBS)
