Index: flex-2.5.35/Makefile.in
===================================================================
--- flex-2.5.35.orig/Makefile.in	2011-04-20 15:30:27.849413788 +0400
+++ flex-2.5.35/Makefile.in	2011-04-20 18:50:07.925252127 +0400
@@ -274,8 +274,7 @@
 m4 = @M4@
 indent = @INDENT@
 lib_LIBRARIES = \
-	libfl.a \
-	libfl_pic.a
+	$(SONAME)
 
 flex_SOURCES = \
 	ccl.c \
@@ -310,6 +309,7 @@
 
 libfl_pic_a_CFLAGS = \
 	-fPIC         \
+	-DPIC         \
 	$(AM_CFLAGS)
 
 noinst_HEADERS = \
@@ -457,14 +457,14 @@
 	    $(libLIBRARIES_INSTALL) "$$p" "$(DESTDIR)$(libdir)/$$f"; \
 	  else :; fi; \
 	done
-	@$(POST_INSTALL)
-	@list='$(lib_LIBRARIES)'; for p in $$list; do \
-	  if test -f $$p; then \
-	    p=$(am__strip_dir) \
-	    echo " $(RANLIB) '$(DESTDIR)$(libdir)/$$p'"; \
-	    $(RANLIB) "$(DESTDIR)$(libdir)/$$p"; \
-	  else :; fi; \
-	done
+#	@$(POST_INSTALL)
+#	@list='$(lib_LIBRARIES)'; for p in $$list; do \
+#	  if test -f $$p; then \
+#	    p=$(am__strip_dir) \
+#	    echo " $(RANLIB) '$(DESTDIR)$(libdir)/$$p'"; \
+#	    $(RANLIB) "$(DESTDIR)$(libdir)/$$p"; \
+#	  else :; fi; \
+#	done
 
 uninstall-libLIBRARIES:
 	@$(NORMAL_UNINSTALL)
@@ -484,6 +484,8 @@
 	-rm -f libfl_pic.a
 	$(libfl_pic_a_AR) libfl_pic.a $(libfl_pic_a_OBJECTS) $(libfl_pic_a_LIBADD)
 	$(RANLIB) libfl_pic.a
+$(SONAME): $(libfl_pic_a_OBJECTS) $(libfl_pic_a_DEPENDENCIES)
+	$(CC) -shared $(CFLAGS) -G -h $(SONAME) -o $(SONAME) $(libfl_pic_a_OBJECTS) $(libfl_pic_a_LIBADD)
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
