Index: pastedeploy-1.5.0/paste/deploy/paster_templates.py
===================================================================
--- pastedeploy-1.5.0.orig/paste/deploy/paster_templates.py	2011-07-19 12:44:57.537413892 +0400
+++ pastedeploy-1.5.0/paste/deploy/paster_templates.py	2011-07-19 12:44:59.613943351 +0400
@@ -9,7 +9,7 @@
 
 class PasteDeploy(Template):
 
-    _template_dir = 'paster_templates/paste_deploy'
+    _template_dir = '/usr/share/paster_templates/paste_deploy'
     summary = "A web application deployed through paste.deploy"
 
     egg_plugins = ['PasteDeploy']
