--- apt-0.8.10.3+squeeze1.orig/doc/po4a.conf
+++ apt-0.8.10.3+squeeze1/doc/po4a.conf
@@ -9,6 +9,7 @@
 [type: entity]  apt.ent $lang:$lang/apt.ent
 [type: docbook] apt-cache.8.xml $lang:$lang/apt-cache.$lang.8.xml add_$lang:xml.add
 [type: docbook] apt-cdrom.8.xml $lang:$lang/apt-cdrom.$lang.8.xml add_$lang:xml.add
+[type: docbook] apt-clone.8.xml $lang:$lang/apt-clone.$lang.8.xml add_$lang:xml.add
 [type: docbook] apt-config.8.xml $lang:$lang/apt-config.$lang.8.xml add_$lang:xml.add
 [type: docbook] apt-extracttemplates.1.xml $lang:$lang/apt-extracttemplates.$lang.1.xml add_$lang:xml.add
 [type: docbook] apt-ftparchive.1.xml $lang:$lang/apt-ftparchive.$lang.1.xml add_$lang:xml.add
