Index: python-lzma-0.5.3/setup.py
===================================================================
--- python-lzma-0.5.3.orig/setup.py	2011-08-17 15:58:30.135765972 +0400
+++ python-lzma-0.5.3/setup.py	2011-08-17 15:59:48.380362463 +0400
@@ -40,7 +40,7 @@
 for i in xrange(len(c_files)):
     c_files[i] = os.path.join('src', c_files[i])
 
-compile_args = []
+compile_args = ['-D_XPG6']
 warnflags = ['-Wall', '-Wextra', '-pedantic', '-Wswitch-enum', '-Wswitch-default']
 compile_args.extend(warnflags)
 link_args = []
