--- ./adduser.bak	2011-07-19 15:06:24.153567772 +0100
+++ ./adduser	2011-07-19 15:11:03.113480221 +0100
@@ -437,8 +437,8 @@
     &systemcall($useradd, '-d', $home_dir, '-g', $ingroup_name, '-s',
 		$shell, '-u', $new_uid, $new_name);
     if(!$disabled_login) {
-        my $usermod = &which('usermod');
-        &systemcall($usermod, '-p', '*', $new_name);
+        my $passwd = &which('passwd');
+        &systemcall($passwd, '-l', $new_name);
     }
     my $chage = &which('chage');
     print "$chage -M 99999 $new_name\n" if ($verbose > 1);
@@ -562,8 +562,8 @@
 	}
     } else {
 	if(!$disabled_login) {
-           my $usermod = &which('usermod');
-	    &systemcall($usermod, '-p', '*', $new_name);
+           my $passwd = &which('passwd');
+	    &systemcall($passwd, '-l', $new_name);
 	}
     }
 
