--- libevent-2.0.21-stable.orig/configure	2012-11-19 07:13:15.000000000 -0800
+++ libevent-2.0.21-stable/configure	2013-07-15 19:10:18.252149254 -0700
@@ -11120,7 +11120,7 @@
   need_lib_prefix=no
   need_version=no
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
-  soname_spec='${libname}${release}${shared_ext}$major'
+  soname_spec='${libname}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
   hardcode_into_libs=yes
