Index: b/Modules/_ssl.c
===================================================================
--- a/Modules/_ssl.c	2011-08-25 20:59:09.953048459 +0400
+++ b/Modules/_ssl.c	2011-08-25 20:59:42.713034091 +0400
@@ -303,7 +303,14 @@
     else if (proto_version == PY_SSL_VERSION_SSL3)
         self->ctx = SSL_CTX_new(SSLv3_method()); /* Set up context */
     else if (proto_version == PY_SSL_VERSION_SSL2)
+#ifdef OPENSSL_NO_SSL2
+    {
+        errstr = ERRSTR("SSLv2 support not provided anymore on Debian");
+        goto fail;
+    }
+#else
         self->ctx = SSL_CTX_new(SSLv2_method()); /* Set up context */
+#endif
     else if (proto_version == PY_SSL_VERSION_SSL23)
         self->ctx = SSL_CTX_new(SSLv23_method()); /* Set up context */
     PySSL_END_ALLOW_THREADS
