Author: Werner Fink <werner@suse.de>
Description: Updates (and re-enables) awk magic.

diff -Naurp file.orig/magic/Magdir/commands file/magic/Magdir/commands
--- file.orig/magic/Magdir/commands	2010-01-30 09:44:26.000000000 +0000
+++ file/magic/Magdir/commands	2010-02-01 11:51:38.000000000 +0000
@@ -50,6 +50,7 @@
 # update to distinguish from *.vcf files
 # this is broken because postscript has /EBEGIN{ for example.
 #0	search/Ww	BEGIN {			awk script text
+0	regex		=^\\s*BEGIN\\s*[{]	awk script text
 
 # AT&T Bell Labs' Plan 9 shell
 0	string/w	#!\ /bin/rc	Plan 9 rc shell script text executable
