Author: Daniel Baumann <daniel@debian.org>
Description: Adding another shebang path for bash.

diff -Naurp file.orig/magic/Magdir/commands file/magic/Magdir/commands
--- file.orig/magic/Magdir/commands	2010-01-30 09:44:26.000000000 +0000
+++ file/magic/Magdir/commands	2010-02-01 11:54:51.000000000 +0000
@@ -57,6 +57,10 @@
 # bash shell magic, from Peter Tobias (tobias@server.et-inf.fho-emden.de)
 0	string/w	#!\ /bin/bash	Bourne-Again shell script text executable
 !:mime	text/x-shellscript
+0	string/w	#!\ /usr/bin/bash	Bourne-Again shell script text executable
+!:mime	text/x-shellscript
+0	string/w	#!\ /usr/local/bash	Bourne-Again shell script text executable
+!:mime	text/x-shellscript
 0	string/w	#!\ /usr/local/bin/bash	Bourne-Again shell script text executable
 !:mime	text/x-shellscript
 
