Author: Werner Fink <werner@suse.de>
Description: Updating reiserfs magic to detect ReiserFS V3.6.19.

diff -Naurp file.orig/magic/Magdir/filesystems file/magic/Magdir/filesystems
--- file.orig/magic/Magdir/filesystems	2008-05-04 14:12:49.000000000 +0000
+++ file/magic/Magdir/filesystems	2008-05-21 09:43:04.000000000 +0000
@@ -1178,6 +1178,7 @@
 # reiserfs - russell@coker.com.au
 0x10034		string	ReIsErFs	ReiserFS V3.5
 0x10034		string	ReIsEr2Fs	ReiserFS V3.6
+0x10034		string	ReIsEr3Fs	ReiserFS V3.6.19
 >0x1002c 	leshort	x		block size %d
 >0x10032	leshort	&2		(mounted or unclean)
 >0x10000	lelong	x		num blocks %d
