Author: n/a
Description: Updating zip magic.

diff -Naurp file.orig/magic/Magdir/msdos file/magic/Magdir/msdos
--- file.orig/magic/Magdir/msdos	2008-05-31 22:17:03.000000000 +0200
+++ file/magic/Magdir/msdos	2008-06-13 09:44:56.000000000 +0200
@@ -256,7 +256,7 @@
 
 # a few unknown ZIP sfxes, no idea if they are needed or if they are
 # already captured by the generic patterns above
->122		string		Windows\ self-extracting\ ZIP	\b, ZIP self-extracting archive
+>0x7a		string		Windows\ self-extracting\ ZIP	\b, ZIP self-extracting archive
 >(8.s*16)	search/0x20	PKSFX \b, ZIP self-extracting archive (PKZIP)
 # TODO: how to add this? >FileSize-34 string Windows\ Self-Installing\ Executable \b, ZIP self-extracting archive
 #
