Author: n/a
Description: Update RealMedia magic.

diff -Naurp file.orig/magic/Magdir/audio file/magic/Magdir/audio
--- file.orig/magic/Magdir/audio	2008-05-31 22:17:03.000000000 +0200
+++ file/magic/Magdir/audio	2008-06-13 09:48:01.000000000 +0200
@@ -116,7 +116,7 @@
 # Real Audio (Magic .ra\0375)
 0	belong		0x2e7261fd	RealAudio sound file
 !:mime	audio/x-pn-realaudio
-0	string		.RMF		RealMedia file
+0	string		.RMF\0\0\0	RealMedia file
 !:mime	application/vnd.rn-realmedia
 #video/x-pn-realvideo
 #video/vnd.rn-realvideo
