Author: Eric Cooper <ecc@cmu.edu>
Description: Updating ocaml magic (Closes: #488992).

diff -Naurp file.orig/magic/Magdir/ocaml file/magic/Magdir/ocaml
--- file.orig/magic/Magdir/ocaml	2008-07-15 14:13:20.000000000 +0000
+++ file/magic/Magdir/ocaml	2008-07-15 14:13:08.000000000 +0000
@@ -1,7 +1,7 @@
 
 #------------------------------------------------------------------------------
 # ocaml: file(1) magic for Objective Caml files.
-0	string	Caml1999	Objective caml
+0	string	Caml1999	OCaml
 >8	string	X		exec file
 >8	string	I		interface file (.cmi)
 >8	string	O		object file (.cmo)
@@ -10,4 +10,4 @@
 >8	string	Z		native library file (.cmxa)
 >8	string	M		abstract syntax tree implementation file
 >8	string	N		abstract syntax tree interface file
->9	string	>\0		(Version %3.3s).
+>9	string	>\0		(Version %3.3s)
