Author: Toeroek Edwin <edwintorok@gmail.com>
Description: Updating llvm magic (Closes: #505805).

diff -Naurp file.orig/magic/Magdir/llvm file/magic/Magdir/llvm
--- file.orig/magic/Magdir/llvm	2008-10-19 14:31:15.000000000 +0000
+++ file/magic/Magdir/llvm	2008-11-15 19:15:51.000000000 +0000
@@ -1,10 +1,12 @@
 
 #------------------------------------------------------------------------------
 # llvm:  file(1) magic for LLVM byte-codes
-# URL:  http://llvm.cs.uiuc.edu/docs/BytecodeFormat.html#signature
+# URL:  http://llvm.org/docs/BitCodeFormat.html
 # From: Al Stone <ahs3@fc.hp.com>
 
 0	string	llvm	LLVM byte-codes, uncompressed
 0	string	llvc0	LLVM byte-codes, null compression
 0	string	llvc1	LLVM byte-codes, gzip compression
 0	string	llvc2	LLVM byte-codes, bzip2 compression
+0	string	\xde\xc0\x17\x0b	LLVM bitcode, wrapper
+0       string	BC\xc0\xde	LLVM bitcode
