Author: Daniel Baumann <daniel@debian.org>
Description: Adding mime type for 7-zip (Closes: #552742).

diff -Naurp file.orig/magic/Magdir/compress file/magic/Magdir/compress
--- file.orig/magic/Magdir/compress	2009-12-05 17:30:09.000000000 +0100
+++ file/magic/Magdir/compress	2009-12-05 18:54:52.000000000 +0100
@@ -187,6 +187,7 @@
 0	string		7z\274\257\047\034	7-zip archive data,
 >6	byte		x			version %d
 >7	byte		x			\b.%d
+!:mime	application/x-7z-compressed
 
 # Type: LZMA
 # URL:  http://www.7-zip.org/sdk.html
