Author: Edward Betts <edward@debian.org>
Description: New magic for pmenu (Closes: #38543).

diff -Naurp file.orig/magic/Magdir/commands file/magic/Magdir/commands
--- file.orig/magic/Magdir/commands	2007-06-01 19:40:26.000000000 +0000
+++ file/magic/Magdir/commands	2008-01-12 17:34:41.000000000 +0000
@@ -55,3 +55,8 @@
 0	string		Zend\x00		PHP script Zend Optimizer data
 
 0	string		$!			DCL command file
+
+# Type: Pdmenu
+# URL:  http://packages.debian.org/pdmenu
+# From: Edward Betts <edward@debian.org>
+0	string	#!/usr/bin/pdmenu	Pdmenu configuration file text
