Author: Josh Triplett <josh@freedesktop.org>
Description: New magic for SNES SPC700 sound files (Closes: #410847).

diff -Naurp file.orig/magic/Magdir/audio file/magic/Magdir/audio
--- file.orig/magic/Magdir/audio	2008-11-07 15:44:13.000000000 +0000
+++ file/magic/Magdir/audio	2009-02-18 11:43:50.000000000 +0000
@@ -306,6 +306,15 @@
 >122	byte&0x1	=1		PAL
 >122	byte&0x1	=0		NTSC
 
+# Type: SNES SPC700 sound files
+# From: Josh Triplett <josh@freedesktop.org>
+0	string	SNES-SPC700\ Sound\ File\ Data\ v	SNES SPC700 sound file
+>&0	string	0.30					\b, version %s
+>>0x23	byte	0x1B					\b, without ID666 tag
+>>0x23	byte	0x1A					\b, with ID666 tag
+>>>0x2E	string	>\0					\b, song "%.32s"
+>>>0x4E	string	>\0					\b, game "%.32s"
+
 # Impulse tracker module (audio/x-it)
 0	string		IMPM		Impulse Tracker module sound data -
 !:mime	audio/x-mod
