Author: Nicolas Collignon <tsointsoin@gmail.com>
Description: New magic for OpenSSH key files (Closes: #439537).

diff -Naurp file.orig/magic/Magdir/ssh file/magic/Magdir/ssh
--- file.orig/magic/Magdir/ssh	1970-01-01 00:00:00.000000000 +0000
+++ file/magic/Magdir/ssh	2009-02-18 11:50:34.000000000 +0000
@@ -0,0 +1,8 @@
+# Type:	OpenSSH key files
+# From:	Nicolas Collignon <tsointsoin@gmail.com>
+
+0	string	SSH\ PRIVATE\ KEY	OpenSSH RSA1 private key,
+>28	string	>\0			version %s
+
+0	string	ssh-dss\		OpenSSH DSA public key
+0	string	ssh-rsa\		OpenSSH RSA public key
diff -Naurp file.orig/magic/Makefile.am file/magic/Makefile.am
--- file.orig/magic/Makefile.am	2009-02-18 11:49:05.000000000 +0000
+++ file/magic/Makefile.am	2009-02-18 11:51:14.000000000 +0000
@@ -183,6 +183,7 @@ $(MAGIC_FRAGMENT_DIR)/softquad \
 $(MAGIC_FRAGMENT_DIR)/spec \
 $(MAGIC_FRAGMENT_DIR)/spectrum \
 $(MAGIC_FRAGMENT_DIR)/sql \
+$(MAGIC_FRAGMENT_DIR)/ssh \
 $(MAGIC_FRAGMENT_DIR)/sun \
 $(MAGIC_FRAGMENT_DIR)/sysex \
 $(MAGIC_FRAGMENT_DIR)/teapot \
diff -Naurp file.orig/magic/Makefile.in file/magic/Makefile.in
--- file.orig/magic/Makefile.in	2009-02-18 11:49:05.000000000 +0000
+++ file/magic/Makefile.in	2009-02-18 11:51:20.000000000 +0000
@@ -345,6 +345,7 @@ $(MAGIC_FRAGMENT_DIR)/softquad \
 $(MAGIC_FRAGMENT_DIR)/spec \
 $(MAGIC_FRAGMENT_DIR)/spectrum \
 $(MAGIC_FRAGMENT_DIR)/sql \
+$(MAGIC_FRAGMENT_DIR)/ssh \
 $(MAGIC_FRAGMENT_DIR)/sun \
 $(MAGIC_FRAGMENT_DIR)/sysex \
 $(MAGIC_FRAGMENT_DIR)/teapot \
