Author: Benoit Sibaud <bsibaud@april.org>
Description: New magic for TokyoCabinet database (Closes: #481768).

diff -Naurp file.orig/magic/Magdir/database file/magic/Magdir/database
--- file.orig/magic/Magdir/database	2009-05-03 08:22:43.000000000 +0000
+++ file/magic/Magdir/database	2009-05-03 08:23:25.000000000 +0000
@@ -272,3 +272,9 @@
 # From:	Benoit Sibaud <bsibaud@april.org>
 0	string		\\[depot\\]\n\f		Quick Database Manager, little endian
 0	string		\\[DEPOT\\]\n\f		Quick Database Manager, big endian
+
+# Type:	TokyoCabinet database
+# URL:	http://tokyocabinet.sourceforge.net/
+# From:	Benoit Sibaud <bsibaud@april.org>
+0	string		ToKyO\ CaBiNeT\n	TokyoCabinet database
+>14	string		x			(version %s)
