Author: Daniel Novotny <dnovotny@redhat.com>
Description: Add new magic for xfs dumps.

diff -Naurp file.orig/magic/Magdir/filesystems file/magic/Magdir/filesystems
--- file.orig/magic/Magdir/filesystems	2009-12-05 17:44:07.000000000 +0100
+++ file/magic/Magdir/filesystems	2009-12-05 17:46:52.000000000 +0100
@@ -1414,3 +1414,10 @@
 >40	ulelong	>0	\b fblock size = %ld,
 >44	ulelong	>0	\b block size = %ld,
 >48	ulequad	>0	\b bytes = %lld
+
+# Type:	xfs metadump image
+# From:	Daniel Novotny <dnovotny@redhat.com>
+# mb_magic XFSM at 0; superblock magic XFSB at 1 << mb_blocklog
+# but can we do the << ? For now it's always 512 (0x200) anyway.
+0	string	XFSM
+>0x200	string	XFSB	XFS filesystem metadump image
