Author: Daniel Novotny <dnovotny@redhat.com>
Description: Add new magic for delta iso files.

diff -Naurp file.orig/magic/Magdir/filesystems file/magic/Magdir/filesystems
--- file.orig/magic/Magdir/filesystems	2009-12-05 17:53:02.000000000 +0100
+++ file/magic/Magdir/filesystems	2009-12-05 17:54:22.000000000 +0100
@@ -1421,3 +1421,8 @@
 # but can we do the << ? For now it's always 512 (0x200) anyway.
 0	string	XFSM
 >0x200	string	XFSB	XFS filesystem metadump image
+
+# Type:	delta ISO
+# From:	Daniel Novotny <dnovotny@redhat.com>
+0	string	DISO	Delta ISO data,
+>4	belong	x	version %d
