Author: Daniel Novotny <dnovotny@redhat.com>
Description: Add new magic for delta rpm files.

diff -Naurp file.orig/magic/Magdir/rpm file/magic/Magdir/rpm
--- file.orig/magic/Magdir/rpm	2009-12-05 17:30:09.000000000 +0100
+++ file/magic/Magdir/rpm	2009-12-05 17:56:35.000000000 +0100
@@ -21,3 +21,9 @@
 >>8	beshort		11		MIPSel
 >>8	beshort		12		ARM
 >>10	string		x		%s
+
+# Type:	Delta RPM
+# From:	Daniel Novotny (dnovotny@redhat.com)
+0	string		drpm		Delta RPM
+!:mime application/x-rpm
+>12	string		x		%s
