Author: Daniel Baumann <daniel@debian.org>
Description: Adding new magics for Chiasmus (Closes: #540368).

diff -Naurp file.orig/magic/Magdir/bsi file/magic/Magdir/bsi
--- file.orig/magic/Magdir/bsi	1970-01-01 01:00:00.000000000 +0100
+++ file/magic/Magdir/bsi	2009-12-05 18:16:59.000000000 +0100
@@ -0,0 +1,9 @@
+# Chiasmus is a encryption standard developed by the German Federal
+# Office for Information Security (Bundesamt fuer Sicherheit in der
+# Informationstechnik).
+
+# Extension: .xia
+0	string	XIA1	Chiasmus encrypted data
+
+# Extension: .xis
+0	string	XIS	Chiasmus key
diff -Naurp file.orig/magic/Makefile.am file/magic/Makefile.am
--- file.orig/magic/Makefile.am	2009-12-05 18:09:09.000000000 +0100
+++ file/magic/Makefile.am	2009-12-05 18:17:20.000000000 +0100
@@ -30,6 +30,7 @@ $(MAGIC_FRAGMENT_DIR)/blender \
 $(MAGIC_FRAGMENT_DIR)/blit \
 $(MAGIC_FRAGMENT_DIR)/bout \
 $(MAGIC_FRAGMENT_DIR)/bsdi \
+$(MAGIC_FRAGMENT_DIR)/bsi \
 $(MAGIC_FRAGMENT_DIR)/btsnoop \
 $(MAGIC_FRAGMENT_DIR)/c-lang \
 $(MAGIC_FRAGMENT_DIR)/c64 \
diff -Naurp file.orig/magic/Makefile.in file/magic/Makefile.in
--- file.orig/magic/Makefile.in	2009-12-05 18:09:09.000000000 +0100
+++ file/magic/Makefile.in	2009-12-05 18:17:24.000000000 +0100
@@ -195,6 +195,7 @@ $(MAGIC_FRAGMENT_DIR)/blender \
 $(MAGIC_FRAGMENT_DIR)/blit \
 $(MAGIC_FRAGMENT_DIR)/bout \
 $(MAGIC_FRAGMENT_DIR)/bsdi \
+$(MAGIC_FRAGMENT_DIR)/bsi \
 $(MAGIC_FRAGMENT_DIR)/btsnoop \
 $(MAGIC_FRAGMENT_DIR)/c-lang \
 $(MAGIC_FRAGMENT_DIR)/c64 \
