Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: Add new magic for datafork fonts (Closes: #291908).

diff -Naurp file.orig/magic/Magdir/macintosh file/magic/Magdir/macintosh
--- file.orig/magic/Magdir/macintosh	2010-07-11 00:17:48.692930622 +0200
+++ file/magic/Magdir/macintosh	2010-07-13 17:26:21.169673589 +0200
@@ -376,3 +376,15 @@
 
 # From: Remi Mommsen <mommsen@slac.stanford.edu>
 0		string		BOMStore	Mac OS X bill of materials (BOM) file
+
+# From: Adam Buchbinder <adam.buchbinder@gmail.com>
+# URL: http://en.wikipedia.org/wiki/Datafork_TrueType
+# Derived from the 'fondu' and 'ufond' source code (fondu.sf.net). 'sfnt' is
+# TrueType; 'POST' is PostScript. 'FONT' and 'NFNT' sometimes appear, but I
+# don't know what they mean.
+0	belong	0x100
+>(0x4.L+24)	beshort	x
+>>&4	belong	0x73666e74	Mac OSX datafork font, TrueType
+>>&4	belong	0x464f4e54	Mac OSX datafork font, 'FONT'
+>>&4	belong	0x4e464e54	Mac OSX datafork font, 'NFNT'
+>>&4	belong	0x504f5354	Mac OSX datafork font, PostScript
