Author: Євгеній Мещеряков <eugen@debian.org>
Description: Adding new magics for GDSII (Closes: #576462).

diff -Naurp file.orig/magic/Magdir/scientific file/magic/Magdir/scientific
--- file.orig/magic/Magdir/scientific	2010-07-13 19:20:16.702717178 +0200
+++ file/magic/Magdir/scientific	2010-07-13 19:19:47.037693749 +0200
@@ -97,3 +97,10 @@
 >>>>&0	regex/1	[A-Z0-9]{4}	Protein Data Bank data, ID Code %s
 !:mime	chemical/x-pdb
 >>>>0	regex/1	[0-9]{2}-[A-Z]{3}-[0-9]{2}	\b, %s
+
+# Type:	GDSII Stream file
+0	belong	0x00060002	GDSII Stream file
+>4	byte	0x00
+>>5	byte	x		version %d.0
+>4	byte	>0x00		version %d
+>>5	byte	x		\b.%d
