Author: Michael Piefel <piefel@debian.org>
Description: n/a

diff -Naurp file.orig/src/Makefile.in file/src/Makefile.in
--- file.orig/src/Makefile.in	2008-01-12 17:55:24.000000000 +0000
+++ file/src/Makefile.in	2008-01-12 17:56:46.000000000 +0000
@@ -300,7 +300,7 @@ clean-binPROGRAMS:
 	done
 file$(EXEEXT): $(file_OBJECTS) $(file_DEPENDENCIES) 
 	@rm -f file$(EXEEXT)
-	$(LINK) $(file_OBJECTS) $(file_LDADD) $(LIBS)
+	$(LINK) $(file_OBJECTS) $(file_LDADD)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
