--- iometer-1.1.0-rc1/src/Makefile-Solaris.i386	2010-11-30 02:13:41.000000000 +0300
+++ iometer-1.1.0-rc1/src/Makefile-Solaris.i386	2011-06-24 23:17:09.846248429 +0400
@@ -88,7 +88,8 @@
 	-DIOMTR_CPU_I386 \
         -D_LARGEFILE64_SOURCE \
         -D_REENTRANT \
-	-DUSE_NEW_DISCOVERY_MECHANISM -DFORCE_STRUCT_ALIGN
+	-DUSE_NEW_DISCOVERY_MECHANISM -DFORCE_STRUCT_ALIGN \
+	-DNOMINMAX
 #-I/opt/SUNWspro/include/CC \
 # CFLAGS=-compat=4
 
--- iometer-1.1.0-rc1/src/pack.h	2010-11-30 02:13:41.000000000 +0300
+++ iometer-1.1.0-rc1/src/pack.h	2011-06-24 23:29:56.873815011 +0400
@@ -61,7 +61,7 @@
 #elif defined(IOMTR_OSFAMILY_UNIX)
 
 #ifndef GCC_ATTRIBUTE_ALIGN
-#pragma pack(push, 1)
+#pragma pack(1)
 #endif //GCC_ATTRIBUTE_ALIGN
 
 // Look for GCC_ATTRIBUTE_ALIGN in iocommon.h for this definition
--- iometer-1.1.0-rc1/src/unpack.h	2010-11-30 02:13:41.000000000 +0300
+++ iometer-1.1.0-rc1/src/unpack.h	2011-06-24 23:30:06.894678416 +0400
@@ -63,7 +63,7 @@
 
 // Nothing to do for the GCC_ATTRIBUTE_ALIGN case
 #ifndef GCC_ATTRIBUTE_ALIGN
-#pragma pack(pop)
+#pragma pack(1)
 #endif //GCC_ATTRIBUTE_ALIGN
 
 // Look for GCC_ATTRIBUTE_ALIGN in iocommon.h for this definition
