Author: n/a
Description: Updating Digifax magic.

diff -Naurp file.orig/magic/Magdir/modem file/magic/Magdir/modem
--- file.orig/magic/Magdir/modem	2008-06-13 15:00:30.000000000 +0200
+++ file/magic/Magdir/modem	2008-06-13 15:25:40.000000000 +0200
@@ -2,9 +2,9 @@
 # modem:  file(1) magic for modem programs
 #
 # From: Florian La Roche <florian@knorke.saar.de>
-4	string		Research,	Digifax-G3-File
->29	byte		1		, fine resolution
->29	byte		0		, normal resolution
+1	string		PC\ Research,\ Inc	Digifax-G3-File
+>29	byte		1			\b, fine resolution
+>29	byte		0			\b, normal resolution
 
 0	short		0x0100		raw G3 data, byte-padded
 0	short		0x1400		raw G3 data
