Author: n/a
Description: Updating Mono magic.

diff -Naurp file.orig/magic/Magdir/msdos file/magic/Magdir/msdos
--- file.orig/magic/Magdir/msdos	2008-07-02 15:16:52.000000000 +0000
+++ file/magic/Magdir/msdos	2008-07-15 13:33:43.000000000 +0000
@@ -68,6 +68,7 @@
 >>&0	leshort		0x290	PA-RISC
 >>&18	leshort&0x0100	>0	32-bit
 >>&18	leshort&0x1000	>0	system file
+>>&228	lelong		>0	\b, Mono/.Net assembly
 >>&0xf4 search/0x140 \x0\x40\x1\x0
 >>>(&0.l+(4)) string MSCF \b, WinHKI CAB self-extracting archive
 >30		string	Copyright\ 1989-1990\ PKWARE\ Inc.	Self-extracting PKZIP archive
