Author: n/a
Description: Updating pfm magic.

diff -Naurp file.orig/magic/Magdir/msdos file/magic/Magdir/msdos
--- file.orig/magic/Magdir/msdos	2008-07-02 15:16:52.000000000 +0000
+++ file/magic/Magdir/msdos	2008-07-15 13:36:26.000000000 +0000
@@ -504,6 +504,13 @@
 # Acroread or something	 files wrongly identified as G3	 .pfm
 # these have the form \000 \001 any? \002 \000 \000
 # or \000 \001 any? \022 \000 \000
+0	belong&0xffff00ff	0x00010012	PFM data
+>4	string			\000\000
+>6	string			>\060		- %s
+
+0	belong&0xffff00ff	0x00010002	PFM data
+>4	string			\000\000
+>6	string			>\060		- %s
 #0	string	\000\001 pfm?
 #>3	string	\022\000\000Copyright\	yes
 #>3	string	\002\000\000Copyright\	yes
