Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: pdating wav magics (Closes: #508174).

diff -Naurp file.orig/magic/Magdir/riff file/magic/Magdir/riff
--- file.orig/magic/Magdir/riff	2008-12-05 17:28:11.000000000 -0500
+++ file/magic/Magdir/riff	2008-12-05 17:33:48.000000000 -0500
@@ -39,12 +39,14 @@
 >>20	leshort		2		\b, Microsoft ADPCM
 >>20	leshort		6		\b, ITU G.711 A-law
 >>20	leshort		7		\b, ITU G.711 mu-law
+>>20	leshort		8		\b, Microsoft DTS
 >>20	leshort		17		\b, IMA ADPCM
 >>20	leshort		20		\b, ITU G.723 ADPCM (Yamaha)
 >>20	leshort		49		\b, GSM 6.10
 >>20	leshort		64		\b, ITU G.721 ADPCM
 >>20	leshort		80		\b, MPEG
 >>20	leshort		85		\b, MPEG Layer 3
+>>20	leshort		0x2001		\b, DTS
 >>22	leshort		=1		\b, mono
 >>22	leshort		=2		\b, stereo
 >>22	leshort		>2		\b, %d channels
