Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: Updating magic for spectrum tap files (Closes: #501589).

diff -Naurp file.orig/magic/Magdir/spectrum file/magic/Magdir/spectrum
--- file.orig/magic/Magdir/spectrum	2009-03-21 09:20:03.000000000 +0000
+++ file/magic/Magdir/spectrum	2009-03-21 17:24:09.000000000 +0000
@@ -19,13 +19,17 @@
 # Tape file. This assumes the .TAP starts with a Spectrum-format header,
 # which nearly all will.
 #
-0       string          \023\000\000    Spectrum .TAP data
->4      string          x               "%-10.10s"
->3      byte            0               - BASIC program
->3      byte            1               - number array
->3      byte            2               - character array
->3      byte            3               - memory block
->>14    belong          0x001B0040      (screen)
+# Update: Sanity-check string contents to be printable.
+#  -Adam Buchbinder <adam.buchbinder@gmail.com>
+#
+0       string          \023\000\000
+>4      string          >\0
+>>4     string          <\177           Spectrum .TAP data "%-10.10s"
+>>>3    byte            0               - BASIC program
+>>>3    byte            1               - number array
+>>>3    byte            2               - character array
+>>>3    byte            3               - memory block
+>>>>14  belong          0x001B0040      (screen)
 
 # The following three blocks are from pak21-spectrum@srcf.ucam.org
 # TZX tape images
