Author: Hugo Graumann <graumann@ucalgary.ca>
Description: Updating tgif magic (Closes: #549601).

diff -Naurp file.orig/magic/Magdir/tgif file/magic/Magdir/tgif
--- file.orig/magic/Magdir/tgif	2009-10-02 12:14:27.000000000 +0000
+++ file/magic/Magdir/tgif	2009-10-04 21:49:54.000000000 +0000
@@ -1,6 +1,5 @@
 #------------------------------------------------------------------------------
 # file(1) magic for tgif(1) files
 # From Hendrik Scholz <hendrik@scholz.net>
-
-0	string	%TGIF\ x 		Tgif file version %s
-
+0	string	%TGIF\			Tgif file version
+>6	string	x			%s
