Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: Update magic of TrueType Collections (Closes: #545709).

diff -Naurp file.orig/magic/Magdir/fonts file/magic/Magdir/fonts
--- file.orig/magic/Magdir/fonts	2009-12-05 18:26:41.000000000 +0100
+++ file/magic/Magdir/fonts	2009-12-05 18:26:58.000000000 +0100
@@ -58,7 +58,15 @@
 0	string		\007\001\001\000Copyright\ (c)\ 199	Adobe Multiple Master font
 0	string		\012\001\001\000Copyright\ (c)\ 199	Adobe Multiple Master font
 
+# TrueType/OpenType font collections (.ttc)
+# http://www.microsoft.com/typography/otspec/otff.htm
 0	string		ttcf		TrueType font collection data
+>4	belong		0x00010000	\b, 1.0
+>>8	belong		>0		\b, %d fonts
+>4	belong		0x00020000	\b, 2.0
+>>8	belong		>0		\b, %d fonts
+# 0x44454947 = 'DSIG'
+>>>16	belong		0x44534947	\b, digitally signed
 
 # Opentype font data from Avi Bercovich
 0	string		OTTO		OpenType font data
