Author: Eloi Notario <entfe001@gmail.com>
Description: Updatng magic for lzma (Closes: #576950).

diff -Naurp file.orig/magic/Magdir/compress file/magic/Magdir/compress
--- file.orig/magic/Magdir/compress	2010-03-27 11:55:00.000000000 +0100
+++ file/magic/Magdir/compress	2010-07-11 00:07:23.645946357 +0200
@@ -190,11 +190,9 @@
 >7	byte		x			\b.%d
 
 # Type: LZMA
-# URL:  http://www.7-zip.org/sdk.html
-# From: Robert Millan <rmh@aybabtu.com> and Reuben Thomas <rrt@sc3d.org>
-# Commented out because apparently not reliable (according to Debian
-# bug #364260)
-#0	string		]\000\000\200\000	LZMA compressed data
+0	lelong		0x8000005d		LZMA compressed data,
+>5	lequad		=0xffffffffffffffff	streamed
+>5	lequad		!0xffffffffffffffff	non-streamed, size %lld
 
 # http://tukaani.org/xz/xz-file-format.txt
 0	ustring		\xFD7zXZ\x00		xz compressed data
