Author: Daniel Baumann <daniel@debian.org>
Description: Updatng magic for xz (Closes: #541087).

diff -Naurp file.orig/magic/Magdir/compress file/magic/Magdir/compress
--- file.orig/magic/Magdir/compress	2010-07-11 02:03:39.853952057 +0200
+++ file/magic/Magdir/compress	2010-07-11 02:04:12.528952297 +0200
@@ -196,7 +196,7 @@
 >5	lequad		!0xffffffffffffffff	non-streamed, size %lld
 
 # http://tukaani.org/xz/xz-file-format.txt
-0	ustring		\xFD7zXZ\x00		xz compressed data
+0	ustring		\xFD7zXZ\x00		XZ compressed data
 !:mime	application/x-xz
 
 # AFX compressed files (Wolfram Kleff)
