Author: Daniel van Eeden <daniel_e@dds.nl>
Description: New magic for PAR2 archive files (Closes: #294070).

diff -Naurp file.orig/magic/Magdir/archive file/magic/Magdir/archive
--- file.orig/magic/Magdir/archive	2010-01-30 09:44:26.000000000 +0000
+++ file/magic/Magdir/archive	2010-02-01 12:09:56.000000000 +0000
@@ -830,3 +830,7 @@
 >24	belong	0		no checksum
 >24	belong	1		SHA-1 checksum
 >24	belong	2		MD5 checksum
+
+# Type: Parity Archive
+# From: Daniel van Eeden <daniel_e@dds.nl>
+0	string	PAR2		Parity Archive Volume Set
