Author: Jelmer Vernooij <jelmer@samba.org>
Description: New magic for Power-Tab

diff -Naurp file.orig/magic/Magdir/chord file/magic/Magdir/chord
--- file.orig/magic/Magdir/chord	2008-02-19 17:27:28.000000000 +0000
+++ file/magic/Magdir/chord	2009-02-18 11:42:42.000000000 +0000
@@ -7,3 +7,8 @@
 #
 0	string		{title		Chord text file
 
+# Type:	PowerTab file format
+# URL:	http://www.power-tab.net/
+# From:	Jelmer Vernooij <jelmer@samba.org>
+0	string		ptab\003\000	Power-Tab v3 Tablature File
+0	string		ptab\004\000	Power-Tab v4 Tablature File
