Author: Sven Hartge <debian@ds9.argh.org>
Description: New magic for ScummVM savegame files (Closes: #263488).

diff -Naurp file.orig/magic/Magdir/console file/magic/Magdir/console
--- file.orig/magic/Magdir/console	2007-09-26 20:22:31.000000000 +0000
+++ file/magic/Magdir/console	2008-01-12 17:36:55.000000000 +0000
@@ -252,3 +252,7 @@
 # .w3m
 0	string HM3W				Warcraft III map file
 
+# Type: scummVM savegame files
+# From: Sven Hartge <debian@ds9.argh.org>
+0	string	SCVM	ScummVM savegame
+>12	string	>\0	"%s"
