Author: Eduardo Sabbatella <eduardo_sabbatella@yahoo.com.ar>
Description: New magic for SGF Smart Game Format

diff -Naurp file.orig/magic/Magdir/games file/magic/Magdir/games
--- file.orig/magic/Magdir/games	2008-04-23 19:00:59.000000000 +0000
+++ file/magic/Magdir/games	2008-07-15 13:41:19.000000000 +0000
@@ -243,3 +243,14 @@
 >&0		string		n\ 					NetImmerse game engine file
 >>&0		regex		[0-9a-z.]+				\b, version %s
 
+# Type:	SGF Smart Game Format
+# URL:	http://www.red-bean.com/sgf/
+# From:	Eduardo Sabbatella <eduardo_sabbatella@yahoo.com.ar>
+2	regex/c	\\(;.*GM\\[[0-9]{1,2}\\]	Smart Game Format
+>2	regex/c	GM\\[1\\]			- Go Game
+>2	regex/c	GM\\[6\\]			- BackGammon Game
+>2	regex/c	GM\\[11\\]			- Hex Game
+>2	regex/c	GM\\[18\\]			- Amazons Game
+>2	regex/c	GM\\[19\\]			- Octi Game
+>2	regex/c	GM\\[20\\]			- Gess Game
+>2	regex/c	GM\\[21\\]			- twix Game
