Author: Nicolas Collignon <tsointsoin@gmail.com>
Description: New magic for OpenSSL certificates/key files (Closes: #439537).

diff -Naurp file.orig/magic/Magdir/ssl file/magic/Magdir/ssl
--- file.orig/magic/Magdir/ssl	1970-01-01 00:00:00.000000000 +0000
+++ file/magic/Magdir/ssl	2009-02-18 11:53:12.000000000 +0000
@@ -0,0 +1,7 @@
+# Type: OpenSSL certificates/key files
+# From: Nicolas Collignon <tsointsoin@gmail.com>
+
+0	string	-----BEGIN\ CERTIFICATE-----	PEM certificate
+0	string	-----BEGIN\ CERTIFICATE\ REQ	PEM certificate request
+0	string	-----BEGIN\ RSA\ PRIVATE	PEM RSA private key
+0	string	-----BEGIN\ DSA\ PRIVATE	PEM DSA private key
diff -Naurp file.orig/magic/Makefile.am file/magic/Makefile.am
--- file.orig/magic/Makefile.am	2009-02-18 11:52:50.000000000 +0000
+++ file/magic/Makefile.am	2009-02-18 11:53:27.000000000 +0000
@@ -184,6 +184,7 @@ $(MAGIC_FRAGMENT_DIR)/spec \
 $(MAGIC_FRAGMENT_DIR)/spectrum \
 $(MAGIC_FRAGMENT_DIR)/sql \
 $(MAGIC_FRAGMENT_DIR)/ssh \
+$(MAGIC_FRAGMENT_DIR)/ssl \
 $(MAGIC_FRAGMENT_DIR)/sun \
 $(MAGIC_FRAGMENT_DIR)/sysex \
 $(MAGIC_FRAGMENT_DIR)/teapot \
diff -Naurp file.orig/magic/Makefile.in file/magic/Makefile.in
--- file.orig/magic/Makefile.in	2009-02-18 11:52:50.000000000 +0000
+++ file/magic/Makefile.in	2009-02-18 11:53:31.000000000 +0000
@@ -346,6 +346,7 @@ $(MAGIC_FRAGMENT_DIR)/spec \
 $(MAGIC_FRAGMENT_DIR)/spectrum \
 $(MAGIC_FRAGMENT_DIR)/sql \
 $(MAGIC_FRAGMENT_DIR)/ssh \
+$(MAGIC_FRAGMENT_DIR)/ssl \
 $(MAGIC_FRAGMENT_DIR)/sun \
 $(MAGIC_FRAGMENT_DIR)/sysex \
 $(MAGIC_FRAGMENT_DIR)/teapot \
