Author: Michael Piefel <piefel@debian.org>
Description: New magic for SVN dumps (closes: #256652).

diff -Naurp file.orig/magic/Magdir/revision file/magic/Magdir/revision
--- file.orig/magic/Magdir/revision	2008-02-19 17:27:33.000000000 +0000
+++ file/magic/Magdir/revision	2009-02-18 11:54:35.000000000 +0000
@@ -17,3 +17,8 @@
 0	string	HG10		Mercurial bundle,
 >4	string	UN		uncompressed
 >4	string	BZ		bzip2 compressed
+
+# Type:	Subversion (SVN) dumps
+# From:	Uwe Zeisberger <zeisberg@informatik.uni-freiburg.de>
+0	string	SVN-fs-dump-format-version:	Subversion dumpfile
+>28	string	>\0				(version: %s)
