Author: Mathias Brodala <info@noctus.net>
Description: New magic for X11 cursor files (Closes: #451246).

diff -Naurp file.orig/magic/Magdir/images file/magic/Magdir/images
--- file.orig/magic/Magdir/images	2009-02-18 11:38:10.000000000 +0000
+++ file/magic/Magdir/images	2009-02-18 11:39:22.000000000 +0000
@@ -610,3 +610,8 @@
 # URL:	http://www.jisyo.com/cgibin/view.cgi?EXT=pe5 (Japanese)
 # From:	Simon Horman <horms@debian.org>
 0	string	IIO2H			Ulead Photo Explorer5
+
+# Type:	X11 cursor
+# URL:	http://webcvs.freedesktop.org/mime/shared-mime-info/freedesktop.org.xml.in?view=markup
+# From:	Mathias Brodala <info@noctus.net>
+0	string	Xcur			X11 cursor
