Author: Werner Fink <werner@suse.de>
Description: Adds new magic for CROM filesystem.

diff -Naurp file.orig/magic/Magdir/filesystems file/magic/Magdir/filesystems
--- file.orig/magic/Magdir/filesystems	2010-02-01 12:11:02.000000000 +0000
+++ file/magic/Magdir/filesystems	2010-02-01 12:12:16.000000000 +0000
@@ -1500,3 +1500,14 @@
 0	string XFSM
 >0x200	string XFSB	XFS filesystem metadump image
 
+# Type:	CROM filesystem
+# From:	Werner Fink <werner@suse.de>
+0	string	CROMFS	CROMFS
+>6	string	>\0	\b version %2.2s,
+>8	ulequad	>0	\b block data at %lld,
+>16	ulequad	>0	\b fblock table at %lld,
+>24	ulequad	>0	\b inode table at %lld,
+>32	ulequad	>0	\b root at %lld,
+>40	ulelong	>0	\b fblock size = %ld,
+>44	ulelong	>0	\b block size = %ld,
+>48	ulequad	>0	\b bytes = %lld
