Author: Alan Woodland <ajw05@aber.ac.uk>
Description: Add support for BLCR context files (Closes: #538407).

diff -Naurp file.orig/magic/Magdir/blcr file/magic/Magdir/blcr
--- file.orig/magic/Magdir/blcr	1970-01-01 00:00:00.000000000 +0000
+++ file/magic/Magdir/blcr	2009-07-25 16:16:17.000000000 +0000
@@ -0,0 +1,25 @@
+# Berkeley Lab Checkpoint Restart (BLCR) checkpoint context files
+# http://ftg.lbl.gov/checkpoint
+0	string	C\0\0\0R\0\0\0	BLCR
+>16	lelong	1	x86
+>16	lelong	3	alpha
+>16	lelong	5	x86-64
+>16	lelong	7	ARM
+>8	lelong	x	context data (little endian, version %d)
+# Uncomment the following only of your "file" program supports "search"
+#>0	search/1024	VMA\06	for kernel
+#>>&1	byte	x	%d.
+#>>&2	byte	x	%d.
+#>>&3	byte	x	%d
+0	string	\0\0\0C\0\0\0R	BLCR
+>16	belong	2	SPARC
+>16	belong	4	ppc
+>16	belong	6	ppc64
+>16	belong	7	ARMEB
+>16	belong	8	SPARC64
+>8	belong	x	context data (big endian, version %d)
+# Uncomment the following only of your "file" program supports "search"
+#>0	search/1024	VMA\06	for kernel
+#>>&1	byte	x	%d.
+#>>&2	byte	x	%d.
+#>>&3	byte	x	%d
diff -Naurp file.orig/magic/Makefile.am file/magic/Makefile.am
--- file.orig/magic/Makefile.am	2009-07-25 16:15:25.000000000 +0000
+++ file/magic/Makefile.am	2009-07-25 16:16:54.000000000 +0000
@@ -25,6 +25,7 @@ $(MAGIC_FRAGMENT_DIR)/att3b \
 $(MAGIC_FRAGMENT_DIR)/audio \
 $(MAGIC_FRAGMENT_DIR)/basis \
 $(MAGIC_FRAGMENT_DIR)/bflt \
+$(MAGIC_FRAGMENT_DIR)/blcr \
 $(MAGIC_FRAGMENT_DIR)/blender \
 $(MAGIC_FRAGMENT_DIR)/blit \
 $(MAGIC_FRAGMENT_DIR)/bout \
diff -Naurp file.orig/magic/Makefile.in file/magic/Makefile.in
--- file.orig/magic/Makefile.in	2009-07-25 16:15:25.000000000 +0000
+++ file/magic/Makefile.in	2009-07-25 16:16:58.000000000 +0000
@@ -190,6 +190,7 @@ $(MAGIC_FRAGMENT_DIR)/att3b \
 $(MAGIC_FRAGMENT_DIR)/audio \
 $(MAGIC_FRAGMENT_DIR)/basis \
 $(MAGIC_FRAGMENT_DIR)/bflt \
+$(MAGIC_FRAGMENT_DIR)/blcr \
 $(MAGIC_FRAGMENT_DIR)/blender \
 $(MAGIC_FRAGMENT_DIR)/blit \
 $(MAGIC_FRAGMENT_DIR)/bout \
