Author: Daniel Baumann <daniel@debian.org>
Description: Adding new magic for Lyx (Closes: #556194).

diff -Naurp file.orig/magic/Magdir/tex file/magic/Magdir/tex
--- file.orig/magic/Magdir/tex	2009-10-02 14:14:27.000000000 +0200
+++ file/magic/Magdir/tex	2009-11-15 11:02:53.000000000 +0100
@@ -91,3 +91,5 @@
 0	search/1	%\ BibTeX\ `	BibTeX custom bibliography style text file
 
 0	search/1	@c\ @mapfile{	TeX font aliases text file
+
+0	string		\#LyX		LyX document text
