Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: Adding new magics for Foveon X3F (Closes: #516800).

diff -Naurp file.orig/magic/Magdir/images file/magic/Magdir/images
--- file.orig/magic/Magdir/images	2009-12-05 18:43:12.000000000 +0100
+++ file/magic/Magdir/images	2009-12-05 18:43:37.000000000 +0100
@@ -655,3 +655,15 @@
 #!mime	image/x-pfs
 >1	regex	[0-9]*\ 		\b, %s
 >>1	regex	\ [0-9]{4}		\bx%s
+
+# Type: Foveon X3F
+# URL:  http://www.photofo.com/downloads/x3f-raw-format.pdf
+# From: Adam Buchbinder <adam.buchbinder@gmail.com>
+# Note that the MIME type isn't defined anywhere that I can find; if
+# there's a canonical type for this format, it should replace this one.
+0	string	FOVb	Foveon X3F raw image data
+!:mime	image/x-x3f
+>6	leshort	x	\b, version %d.
+>4	leshort	x	\b%d
+>28	lelong	x	\b, %dx
+>32	lelong	x	\b%d
