Author: Adam Buchbinder <adam.buchbinder@gmail.com>
Description: Adding new magics for Paint.NET images (Closes: #504779).

diff -Naurp file.orig/magic/Magdir/images file/magic/Magdir/images
--- file.orig/magic/Magdir/images	2009-12-05 18:46:53.000000000 +0100
+++ file/magic/Magdir/images	2009-12-05 18:47:27.000000000 +0100
@@ -667,3 +667,8 @@
 >4	leshort	x	\b%d
 >28	lelong	x	\b, %dx
 >32	lelong	x	\b%d
+
+# Paint.NET file
+# From Adam Buchbinder <adam.buchbinder@gmail.com>
+0	string	PDN3	Paint.NET image data
+!:mime	image/x-paintnet
