Author: n/a
Description: Adding new magics for DACT (Closes: #582945).

diff -Naurp file.orig/magic/Magdir/compress file/magic/Magdir/compress
--- file.orig/magic/Magdir/compress	2010-06-21 00:50:12.608254012 +0200
+++ file/magic/Magdir/compress	2010-06-21 00:53:13.743273541 +0200
@@ -219,3 +219,11 @@
 # URL: http://tukaani.org/xz/
 0	string		\xfd\x37\x7a\x58\x5a\x00	XZ compressed data
 !:mime application/x-xz
+
+# Type:	DACT compressed files
+0	long	0x444354C3	DACT compressed data
+>4	byte	>-1		(version %i.
+>5	byte	>-1		%i.
+>6	byte	>-1		%i)
+>7	long	>0		, original size: %i bytes
+>15	long	>30		, block size: %i bytes
