Author: Frédéric Brière <fbriere@fbriere.net>
Description: Adding new magics for Git index files (Closes: #583679).

diff -Naurp file.orig/magic/Magdir/revision file/magic/Magdir/revision
--- file.orig/magic/Magdir/revision	2010-07-13 19:28:53.070674134 +0200
+++ file/magic/Magdir/revision	2010-07-13 19:28:37.817698688 +0200
@@ -28,6 +28,12 @@
 0	string	\377tOc		Git pack index
 >4	belong	=2		\b, version 2
 
+# Type: Git index file
+# From: Frédéric Brière <fbriere@fbriere.net>
+0	string	DIRC		Git index
+>4	belong	>0		\b, version %d
+>>8	belong	>0		\b, %d entries
+
 # Type:	Mercurial bundles
 # From:	Seo Sanghyeon <tinuviel@sparcs.kaist.ac.kr>
 0	string	HG10		Mercurial bundle,
