Author: Nahuel Greco <ngreco@gmail.com>
Description: Adding new magics for Erlang DETS files (Closes: #589723).

diff -Naurp file.orig/magic/Magdir/erlang file/magic/Magdir/erlang
--- file.orig/magic/Magdir/erlang	2010-07-11 00:17:48.680930074 +0200
+++ file/magic/Magdir/erlang	2010-07-22 00:00:57.484882096 +0200
@@ -17,3 +17,5 @@
 79	string	Tue\ Jan\ 22\ 14:32:44\ MET\ 1991	Erlang JAM file - version 4.2
 
 4	string	1.0\ Fri\ Feb\ 3\ 09:55:56\ MET\ 1995	Erlang JAM file - version 4.3
+
+0	bequad	0x0000000000ABCDEF	Erlang DETS file
