Author: Ori Avtalion <ori@avtalion.name>
Description: Fixing typo and formating issue (Closes: #499754).

diff -Naurp file.orig/doc/magic.man file/doc/magic.man
--- file.orig/doc/magic.man	2010-07-11 00:17:48.664928248 +0200
+++ file/doc/magic.man	2010-07-13 17:08:25.697672263 +0200
@@ -283,7 +283,7 @@ then print the string), with
 The special test
 .Em x
 always evaluates to true.
-.Dv message
+.It Dv message
 The message to be printed if the comparison succeeds.
 If the string contains a
 .Xr printf 3
@@ -348,7 +348,7 @@ If a the test on a line at level
 .Em n
 succeeds, all following tests at level
 .Em n+1
-are performed, and the messages printed if the tests succeed, untile a line
+are performed, and the messages printed if the tests succeed, until a line
 with level
 .Em n
 (or less) appears.
