Index: libxml2-2.7.6/globals.c
===================================================================
--- libxml2-2.7.6.orig/globals.c
+++ libxml2-2.7.6/globals.c
@@ -354,7 +354,7 @@ static const char *xmlTreeIndentStringTh
  * Disabled by default
  */
 int xmlSaveNoEmptyTags = 0;
-static int xmlSaveNoEmptyTagsThrDef = 0;
+int xmlSaveNoEmptyTagsThrDef = 0;
 
 #ifdef LIBXML_SAX1_ENABLED
 /**
Index: libxml2-2.7.6/nanohttp.c
===================================================================
--- libxml2-2.7.6.orig/nanohttp.c
+++ libxml2-2.7.6/nanohttp.c
@@ -168,7 +168,7 @@ static char *proxy = NULL;	 /* the proxy
 static int proxyPort;	/* the proxy port if any */
 static unsigned int timeout = 60;/* the select() timeout in seconds */
 
-static int xmlNanoHTTPFetchContent( void * ctx, char ** ptr, int * len );
+int xmlNanoHTTPFetchContent( void * ctx, char ** ptr, int * len );
 
 /**
  * xmlHTTPErrMemory:
@@ -1788,7 +1788,7 @@ xmlNanoHTTPMimeType( void * ctx ) {
  * -1 if received content length was less than specified or an error 
  * occurred.
  */
-static int
+int
 xmlNanoHTTPFetchContent( void * ctx, char ** ptr, int * len ) {
     xmlNanoHTTPCtxtPtr	ctxt = (xmlNanoHTTPCtxtPtr)ctx;
 
Index: libxml2-2.7.6/xmlunicode.c
===================================================================
--- libxml2-2.7.6.orig/xmlunicode.c
+++ libxml2-2.7.6/xmlunicode.c
@@ -930,8 +930,8 @@ static const xmlChSRange xmlZS[] = {{0x2
     {0x202f, 0x202f}, {0x205f, 0x205f}, {0x3000, 0x3000} };
 static xmlChRangeGroup xmlZG = {9,0,xmlZS,NULL};
 
-static xmlUnicodeNameTable xmlUnicodeBlockTbl = {xmlUnicodeBlocks, 128};
-static xmlUnicodeNameTable xmlUnicodeCatTbl = {xmlUnicodeCats, 36};
+xmlUnicodeNameTable xmlUnicodeBlockTbl = {xmlUnicodeBlocks, 128};
+xmlUnicodeNameTable xmlUnicodeCatTbl = {xmlUnicodeCats, 36};
 
 /**
  * xmlUnicodeLookup:
Index: libxml2-2.7.6/xpath.c
===================================================================
--- libxml2-2.7.6.orig/xpath.c
+++ libxml2-2.7.6/xpath.c
@@ -115,7 +115,7 @@
 double xmlXPathNAN = 0;
 double xmlXPathPINF = 1;
 double xmlXPathNINF = -1;
-static double xmlXPathNZERO = 0; /* not exported from headers */
+double xmlXPathNZERO = 0; /* not exported from headers */
 static int xmlXPathInitialized = 0;
 
 /**
