Index: trousers-0.3.4/src/tspi/tspi_context.c
===================================================================
--- trousers-0.3.4.orig/src/tspi/tspi_context.c	2011-04-14 20:18:04.778141695 +0400
+++ trousers-0.3.4/src/tspi/tspi_context.c	2011-04-14 20:22:55.809554644 +0400
@@ -16,7 +16,6 @@
 #include "trousers/tss.h"
 #include "trousers/trousers.h"
 #include "trousers_types.h"
-#include "trousers_types.h"
 #include "spi_utils.h"
 #include "capabilities.h"
 #include "tsplog.h"
@@ -27,6 +26,8 @@
 #include "tcsd.h"
 #include "obj.h"
 
+/* declaration of function */
+extern TSS_RESULT __tspi_freeTable(TSS_HCONTEXT);
 
 TSS_RESULT
 Tspi_Context_Create(TSS_HCONTEXT * phContext)	/* out */
@@ -55,6 +56,8 @@
 	/* Destroy all objects */
 	obj_close_context(tspContext);
 
+	__tspi_freeTable(tspContext);
+
 	/* close the ps file */
 	PS_close();
 
