Index: sudo-1.7.4p4/configure.in
===================================================================
--- sudo-1.7.4p4.orig/configure.in	2011-06-06 15:42:56.817814279 +0400
+++ sudo-1.7.4p4/configure.in	2011-06-06 15:43:00.000000000 +0400
@@ -2574,7 +2574,7 @@
     if test "$with_ldap" != "yes"; then
 	SUDO_APPEND_LIBPATH(SUDO_LDFLAGS, [${with_ldap}/lib])
 	SUDO_APPEND_LIBPATH(LDFLAGS, [${with_ldap}/lib])
-	CPPFLAGS="${CPPFLAGS} -I${with_ldap}/include"
+	CPPFLAGS="${CPPFLAGS} -I${with_ldap}/include/openldap"
 	with_ldap=yes
     fi
     SUDO_OBJS="${SUDO_OBJS} ldap.o"
@@ -2584,7 +2584,7 @@
     LDAP_LIBS=""
     _LIBS="$LIBS"
     found=no
-    for l in -lldap -llber '-lssl -lcrypto'; do
+    for l in -lldap_r-2.4 -llber '-lssl -lcrypto'; do
 	LIBS="${LIBS} $l"
 	LDAP_LIBS="${LDAP_LIBS} $l"
 	AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <sys/types.h>
@@ -2593,9 +2593,9 @@
     done
     dnl if nothing linked just try with -lldap
     if test "$found" = "no"; then
-	LIBS="${_LIBS} -lldap"
-	LDAP_LIBS="-lldap"
-	AC_MSG_RESULT([not found, using -lldap])
+	LIBS="${_LIBS} -lldap_r-2.4"
+	LDAP_LIBS="-lldap_r-2.4"
+	AC_MSG_RESULT([not found, using -lldap_r-2.4])
     else
 	AC_MSG_RESULT([$LDAP_LIBS])
     fi
Index: sudo-1.7.4p4/configure
===================================================================
--- sudo-1.7.4p4.orig/configure	2011-06-06 15:43:00.000000000 +0400
+++ sudo-1.7.4p4/configure	2011-06-06 15:43:00.000000000 +0400
@@ -17850,7 +17850,7 @@
 	blibpath_add="${blibpath_add}:${with_ldap}/lib"
     fi
 
-	CPPFLAGS="${CPPFLAGS} -I${with_ldap}/include"
+	CPPFLAGS="${CPPFLAGS} -I${with_ldap}/include/openldap"
 	with_ldap=yes
     fi
     SUDO_OBJS="${SUDO_OBJS} ldap.o"
@@ -17861,7 +17861,7 @@
     LDAP_LIBS=""
     _LIBS="$LIBS"
     found=no
-    for l in -lldap -llber '-lssl -lcrypto'; do
+    for l in -lldap_r-2.4 -llber '-lssl -lcrypto'; do
 	LIBS="${LIBS} $l"
 	LDAP_LIBS="${LDAP_LIBS} $l"
 	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -17884,10 +17884,10 @@
     conftest$ac_exeext conftest.$ac_ext
     done
         if test "$found" = "no"; then
-	LIBS="${_LIBS} -lldap"
-	LDAP_LIBS="-lldap"
+	LIBS="${_LIBS} -lldap_r-2.4"
+	LDAP_LIBS="-lldap_r-2.4"
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: not found, using -lldap" >&5
-$as_echo "not found, using -lldap" >&6; }
+$as_echo "not found, using -lldap_r-2.4" >&6; }
     else
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $LDAP_LIBS" >&5
 $as_echo "$LDAP_LIBS" >&6; }
