diff -urN dpkg-1.16.1.1.orig/scripts/dpkg-shlibdeps.pl dpkg-1.16.1.1/scripts/dpkg-shlibdeps.pl
--- dpkg-1.16.1.1.orig/scripts/dpkg-shlibdeps.pl	2011-10-11 22:56:11.000000000 -0700
+++ dpkg-1.16.1.1/scripts/dpkg-shlibdeps.pl	2013-10-23 01:24:33.333404825 -0700
@@ -175,8 +175,8 @@
 	    my $msg = _g("couldn't find library %s needed by %s (ELF " .
 			 "format: '%s'; RPATH: '%s').");
 	    if (scalar(split_soname($soname))) {
-		errormsg($msg, $soname, $file, $obj->{format}, join(":", @{$obj->{RPATH}}));
-		$error_count++;
+		warning($msg, $soname, $file, $obj->{format}, join(":", @{$obj->{RPATH}}));
+#		$error_count++;
 	    } else {
 		warning($msg, $soname, $file, $obj->{format}, join(":", @{$obj->{RPATH}}));
 	    }
@@ -280,9 +280,17 @@
 		    $ignore++ unless scalar(split_soname($soname));
 		    # 3/ when we have been asked to do so
 		    $ignore++ if $ignore_missing_info;
-		    error(_g("no dependency information found for %s " .
-		             "(used by %s)."), $lib, $file)
-		        unless $ignore;
+            if (not $ignore) {
+		        warning(_g("no dependency information found for %s " .
+		             "(used by %s)."), $lib, $file);
+                warning("falling back on `dpkg-query --search'");
+                my $pkg_with_lib = `dpkg-query --search $lib | head -n 1 | cut -d: -f1`;
+                if ($pkg_with_lib) {
+		            $dependencies{$cur_field}{$pkg_with_lib} = Dpkg::Version->new('');
+                } else {
+                    error('No luck :-(');
+                }
+            }
 		}
 	    }
 	}
