Description: Correct minor POD issues in CpantsErrors.pm
Origin: vendor
Forwarded: no
Author: Jonathan Yu <frequency@cpan.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-04-07

--- a/lib/Module/CPANTS/Kwalitee/CpantsErrors.pm
+++ b/lib/Module/CPANTS/Kwalitee/CpantsErrors.pm
@@ -50,11 +50,11 @@
 
 =head1 NAME
 
-Module::CPANTS::Kwalitee::CpantsErrors
+Module::CPANTS::Kwalitee::CpantsErrors - Check for CPANTS testing errors
 
 =head1 SYNOPSIS
 
-Checks if something strange happend during testing
+Checks if something strange happened during testing.
 
 =head1 DESCRIPTION
 
--- a/bin/cpants_lint.pl
+++ b/bin/cpants_lint.pl
@@ -183,7 +183,7 @@
 
 =head3 --no_capture
 
-Turn off capturing of STDOUT and STDERR. Mostly usefull during debugging / development of new features. 
+Turn off capturing of STDOUT and STDERR. Mostly useful during debugging / development of new features. 
 
 If C<--no_capture> is used, the value of C<cpants_error> might be wrong.
 
