Description: Fix --local option for tools/buildhtml.py.
Origin: upstream, http://svn.berlios.de/viewvc/docutils/trunk/docutils/tools/buildhtml.py?r1=6366&r2=6371
Bug: http://sourceforge.net/tracker/?func=detail&aid=2909811&group_id=38414&atid=422030
Last-Update: 2010-07-11

--- a/tools/buildhtml.py
+++ b/tools/buildhtml.py
@@ -181,14 +181,14 @@
         else:
             self.directories = [os.getcwd()]
         for directory in self.directories:
-            try:
-                for root, dirs, files in os.walk(directory):
-                    self.visit(recurse, root, dirs+files)
-            except (AttributeError): # python2.2 does not have os.walk
-                print "no os.walk"
-                os.path.walk(directory, self.visit, recurse)
+            for root, dirs, files in os.walk(directory):
+                # os.walk by default this recurses down the tree,
+                # influence by modifying dirs.
+                if not recurse:
+                    del dirs[:]
+                self.visit(root, dirs+files)
 
-    def visit(self, recurse, directory, names):
+    def visit(self, directory, names):
         settings = self.get_settings('', directory)
         if settings.prune and (os.path.abspath(directory) in settings.prune):
             print >>sys.stderr, '/// ...Skipping directory (pruned):', directory
@@ -211,8 +211,6 @@
                 prune = self.process_txt(directory, name)
                 if prune:
                     break
-        if not recurse:
-            del names[:]
 
     def process_txt(self, directory, name):
         if name.startswith('pep-'):
