# DP: Link _math.o only once to the static library.

Index: b/Modules/Setup.dist
===================================================================
--- a/Modules/Setup.dist	2011-07-26 15:37:13.491067663 +0400
+++ b/Modules/Setup.dist	2011-07-26 15:37:30.054967150 +0400
@@ -169,7 +169,7 @@
 # Modules that should always be present (non UNIX dependent):
 
 #array arraymodule.c	# array objects
-#cmath cmathmodule.c _math.c # -lm # complex math library functions
+#cmath cmathmodule.c # -lm # complex math library functions
 #math mathmodule.c _math.c # -lm # math library functions, e.g. sin()
 #_struct _struct.c	# binary structure packing/unpacking
 #time timemodule.c # -lm # time operations and variables
