# DP: Fix bashism in makesetup shell script

Index: b/Modules/makesetup
===================================================================
--- a/Modules/makesetup	2011-07-26 15:35:00.350048996 +0400
+++ b/Modules/makesetup	2011-07-26 15:37:15.307604509 +0400
@@ -281,7 +281,7 @@
 	-)	;;
 	*)	sedf="@sed.in.$$"
 		trap 'rm -f $sedf' 0 1 2 3
-		echo "1i\\" >$sedf
+		printf "1i\\" >$sedf
 		str="# Generated automatically from $makepre by makesetup."
 		echo "$str" >>$sedf
 		echo "s%_MODOBJS_%$OBJS%" >>$sedf
