--- configure.ac.1	2011-04-15 11:41:42.719715616 +0400
+++ configure.ac	2011-04-15 11:42:04.037331440 +0400
@@ -44,13 +44,13 @@
 m4_include([configure.d/config_project_with_enable])
 m4_include([configure.d/config_os_progs])
 m4_include([configure.d/config_project_paths])
+m4_include([configure.d/config_os_headers])
 m4_include([configure.d/config_project_ipv6_types])
 m4_include([configure.d/config_modules_lib])
 m4_include([configure.d/config_modules_agent])
 m4_include([configure.d/config_os_misc1])
 m4_include([configure.d/config_os_libs])
 m4_include([configure.d/config_project_perl_python])
-m4_include([configure.d/config_os_headers])
 m4_include([configure.d/config_os_misc2])
 m4_include([configure.d/config_project_types])
 m4_include([configure.d/config_os_misc3])
