Index: libtool-2.2.6a/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.2.6a.orig/libltdl/config/ltmain.m4sh	2008-11-18 21:12:39.000000000 +0000
+++ libtool-2.2.6a/libltdl/config/ltmain.m4sh	2008-11-18 21:12:41.000000000 +0000
@@ -5767,6 +5767,9 @@
 	    revision="$number_minor"
 	    lt_irix_increment=no
 	    ;;
+	  *)
+	    func_fatal_configuration "$modename: unknown library version type \`$version_type'"
+	    ;;
 	  esac
 	  ;;
 	no)
