Index: sqlalchemy-0.6.8/setup.py
===================================================================
--- sqlalchemy-0.6.8.orig/setup.py	2011-07-21 02:04:23.600925665 +0400
+++ sqlalchemy-0.6.8/setup.py	2011-07-21 02:04:38.149318525 +0400
@@ -35,7 +35,7 @@
     extra.update(
         features = {'cextensions' : Feature(
             "optional C speed-enhancements",
-            standard = False,
+            standard = True,
             ext_modules = [
                 Extension('sqlalchemy.cprocessors',
                        sources=['lib/sqlalchemy/cextension/processors.c']),
