Description: Use sensible-editor to edit files
Bug-Debian: http://bugs.debian.org/549202
Author: Denis Briand <denis@narcan.fr>
Author: Yury V. Zaytsev <yury@shurup.com>

--- mc-4.7.0.3.orig/lib/util.h	2010-02-26 10:31:38.000000000 +0100
+++ mc-4.7.0.3/lib/util.h	2010-03-17 10:38:58.000000000 +0100
@@ -233,7 +233,7 @@
 #define PATH_ENV_SEP ':'
 #define TMPDIR_DEFAULT "/tmp"
 #define SCRIPT_SUFFIX ""
-#define get_default_editor() "vi"
+#define get_default_editor() "sensible-editor"
 #define OS_SORT_CASE_SENSITIVE_DEFAULT 1
 #define STRCOMP strcmp
 #define STRNCOMP strncmp
