Description: Detect that mc was launched from alternatives symlink
Bug-Debian: http://bugs.debian.org/592208
Author: Yury V. Zaytsev <yury@shurup.com>

--- a/src/main.c
+++ b/src/main.c
@@ -2018,7 +2018,7 @@ mc_main__setup_by_args (int argc, char *argv[])
     base = x_basename (argv[0]);
     tmp = (argc > 0) ? argv[1] : NULL;
 
-    if (!STRNCOMP (base, "mce", 3) || !STRCOMP (base, "vi")) {
+    if (!STRNCOMP (base, "mce", 3) || !STRCOMP (base, "editor") || !STRCOMP (base, "vi")) {
 	edit_one_file = "";
 	if (tmp) {
 	    /*
